diff -urN -x CVS kdemultimedia.orig/akode/arts_plugin/akodePlayObject_impl.cpp kdemultimedia/akode/arts_plugin/akodePlayObject_impl.cpp
--- kdemultimedia.orig/akode/arts_plugin/akodePlayObject_impl.cpp	Sun Sep 12 00:36:45 2004
+++ kdemultimedia/akode/arts_plugin/akodePlayObject_impl.cpp	Tue Oct 12 00:54:01 2004
@@ -356,7 +356,7 @@
                 break;
 	    }
 	}
-        if (buffer->channel_config != MonoStereo || buffer->sample_width > 24 || buffer->sample_width == 0) {
+        if (buffer->channels > 2 || buffer->sample_width > 24 || buffer->sample_width == 0) {
             arts_warning("akode: Incompatible media");
 	    halt();
 	    break;
diff -urN -x CVS kdemultimedia.orig/arts/builder/createtool.cpp kdemultimedia/arts/builder/createtool.cpp
--- kdemultimedia.orig/arts/builder/createtool.cpp	Fri Mar  7 23:11:32 2003
+++ kdemultimedia/arts/builder/createtool.cpp	Mon Oct 18 16:17:31 2004
@@ -156,7 +156,7 @@
 	int input_width = 0;
 	int output_width = 0;
 
-	for(pi = minfo.ports.begin(); pi != minfo.ports.end(); pi++)
+	for(pi = minfo.ports.begin(); pi != minfo.ports.end(); ++pi)
 	{
 		if(pi->direction == Arts::input)
 			input_width++;
@@ -179,7 +179,7 @@
 	int output_width = 0;
 
 	for(pi = minfo.ports.begin(), pni = minfo.portnames.begin();
-		pi != minfo.ports.end(); pi++, pni++)
+		pi != minfo.ports.end(); ++pi, pni++)
 	{
 		StructurePort *port = 0;
 		/*
@@ -209,7 +209,7 @@
 	this->minfo = minfo;
 
 	vector<Arts::PortType>::iterator pi;
-	for(pi = this->minfo.ports.begin(); pi != this->minfo.ports.end(); pi++)
+	for(pi = this->minfo.ports.begin(); pi != this->minfo.ports.end(); ++pi)
 	{
 		/* reverse since we're inside the interface, not outside */
 		if(pi->direction == Arts::input)
@@ -262,7 +262,7 @@
 	std::list<StructureComponent *>::iterator i;
 	std::list<StructureComponent *> *ComponentList = mwidget->structure->getComponentList();
 
-	for(i = ComponentList->begin();i != ComponentList->end();i++)
+	for(i = ComponentList->begin();i != ComponentList->end();++i)
 	{
 		StructureComponent *c = *i;
 		if(c->selected() && !mwidget->hasSpace(c, c->x() + dx, c->y() + dy, true))
@@ -270,7 +270,7 @@
 	}
 
 	mwidget->beginUpdate();
-	for(i = ComponentList->begin();i != ComponentList->end();i++)
+	for(i = ComponentList->begin();i != ComponentList->end();++i)
 	{
 		StructureComponent *c = *i;
 		if(c->selected())
diff -urN -x CVS kdemultimedia.orig/arts/builder/interfacedlg.cpp kdemultimedia/arts/builder/interfacedlg.cpp
--- kdemultimedia.orig/arts/builder/interfacedlg.cpp	Sat Jun 26 12:12:14 2004
+++ kdemultimedia/arts/builder/interfacedlg.cpp	Mon Oct 18 16:17:31 2004
@@ -160,7 +160,7 @@
 	listbox->insertItem((indent + interface).c_str());
 
 	vector<string> *children = Arts::Dispatcher::the()->interfaceRepo().queryChildren(interface);
-	for (vector<string>::iterator ci = children->begin(); ci != children->end(); ci++)
+	for (vector<string>::iterator ci = children->begin(); ci != children->end(); ++ci)
 		update(ci->c_str(), indent+"  ");
 	delete children;
 }
diff -urN -x CVS kdemultimedia.orig/arts/builder/main.cpp kdemultimedia/arts/builder/main.cpp
--- kdemultimedia.orig/arts/builder/main.cpp	Sat Jun 26 12:12:14 2004
+++ kdemultimedia/arts/builder/main.cpp	Mon Oct 18 16:17:31 2004
@@ -110,7 +110,7 @@
 	list<string> datadirs = PortableKDE::globalDirs("data");
 	list<string>::iterator it;
 
-	for(it = datadirs.begin(); it != datadirs.end(); it++)
+	for(it = datadirs.begin(); it != datadirs.end(); ++it)
 	{
 		string common = *it;
 		common += "artsbuilder";
@@ -360,7 +360,7 @@
 
 		vector<Arts::TraderOffer>::iterator i;
 		long n = 1;		/* TODO:PORT: is this necessary? I think not */
-		for(i = offers->begin(); i != offers->end(); i++)
+		for(i = offers->begin(); i != offers->end(); ++i)
 		{
 			Arts::TraderOffer& offer = *i;
 			string name = offer.interfaceName();
diff -urN -x CVS kdemultimedia.orig/arts/builder/menumaker.cpp kdemultimedia/arts/builder/menumaker.cpp
--- kdemultimedia.orig/arts/builder/menumaker.cpp	Thu Nov 29 19:27:49 2001
+++ kdemultimedia/arts/builder/menumaker.cpp	Mon Oct 18 16:17:31 2004
@@ -48,7 +48,7 @@
 
 	list<string>::iterator i;
 
-	for(i=prefixList.begin();i != prefixList.end(); i++)
+	for(i=prefixList.begin();i != prefixList.end(); ++i)
 	{
 		const char *prefix = (*i).c_str();
 
@@ -110,7 +110,7 @@
 {
 	MenuCategory *mc = 0;
 	list<MenuCategory *>::iterator i;
-	for(i=categories.begin();i != categories.end();i++)
+	for(i=categories.begin();i != categories.end();++i)
 	{
 		if((*i)->name() == name) mc = (*i);
 	}
@@ -122,7 +122,7 @@
 	MenuCategory *mc = 0;
 
 	list<MenuCategory *>::iterator i;
-	for(i=categories.begin();i != categories.end();i++)
+	for(i=categories.begin();i != categories.end();++i)
 		if((*i)->matches(name)) mc = (*i);
 	assert(mc);		// root category should always match
 
@@ -130,7 +130,7 @@
 	{
 		// if we hit the root category, it may be better to move the thing
 		// into the catchall category (looks cleaner)
-		for(i=categories.begin();i != categories.end();i++)
+		for(i=categories.begin();i != categories.end();++i)
 			if((*i)->catchall()) mc = (*i);
 	}
 	//mc->menu()->insertItem(name,index); 000
@@ -169,7 +169,7 @@
 void MenuMaker::clear()
 {
 	list<MenuCategory *>::iterator i;
-	for(i=categories.begin();i != categories.end();i++)
+	for(i=categories.begin();i != categories.end();++i)
 	{
 		/* 000
 		KActionMenu *m = (*i)->menu();
diff -urN -x CVS kdemultimedia.orig/arts/builder/module.cpp kdemultimedia/arts/builder/module.cpp
--- kdemultimedia.orig/arts/builder/module.cpp	Wed Aug 20 22:23:34 2003
+++ kdemultimedia/arts/builder/module.cpp	Mon Oct 18 16:17:31 2004
@@ -193,10 +193,10 @@
 	arts_debug("sdfmd...");
 	list<ModulePort *>::iterator i;
 
-	for(i = inports.begin(); i != inports.end(); i++) delete *i;
+	for(i = inports.begin(); i != inports.end(); ++i) delete *i;
 	inports.clear();
 
-	for(i = outports.begin(); i != outports.end(); i++) delete *i;
+	for(i = outports.begin(); i != outports.end(); ++i) delete *i;
 	outports.clear();
 
 	StructureDesc.freeModuleDesc(ModuleDesc);
@@ -422,8 +422,8 @@
 void Module::dumpPorts(list<ModulePort *>& ports)
 {
 	list<ModulePort *>::iterator i;
-	for(i = inports.begin(); i != inports.end(); i++) ports.push_back(*i);
-	for(i = outports.begin(); i != outports.end(); i++) ports.push_back(*i);
+	for(i = inports.begin(); i != inports.end(); ++i) ports.push_back(*i);
+	for(i = outports.begin(); i != outports.end(); ++i) ports.push_back(*i);
 }
 
 QPixmap *Module::pixmap()
diff -urN -x CVS kdemultimedia.orig/arts/builder/mwidget.cpp kdemultimedia/arts/builder/mwidget.cpp
--- kdemultimedia.orig/arts/builder/mwidget.cpp	Tue Feb  3 15:35:53 2004
+++ kdemultimedia/arts/builder/mwidget.cpp	Mon Oct 18 16:17:31 2004
@@ -363,7 +363,7 @@
 
 	std::map<long, ModulePort *> portmap;
 
-	for(pi = portlist.begin(); pi != portlist.end(); pi++)
+	for(pi = portlist.begin(); pi != portlist.end(); ++pi)
 	{
 		ModulePort *port = *pi;
 
@@ -386,10 +386,10 @@
 	vector<long> owner(portlist.size());
 
 	long maxcolor = 0;
-	for(pi = portlist.begin(); pi != portlist.end(); pi++)
+	for(pi = portlist.begin(); pi != portlist.end(); ++pi)
 		color[*pi] = maxcolor++;
 
-	for(pi = portlist.begin(); pi != portlist.end(); pi++)
+	for(pi = portlist.begin(); pi != portlist.end(); ++pi)
 	{
 		ModulePort *src = *pi;
 		unsigned long c;
@@ -418,7 +418,7 @@
 		}
 	}
 
-	for(pi = portlist.begin(); pi != portlist.end(); pi++)
+	for(pi = portlist.begin(); pi != portlist.end(); ++pi)
 	{
 		ModulePort *p = *pi;
 
diff -urN -x CVS kdemultimedia.orig/arts/builder/portposdlg.cpp kdemultimedia/arts/builder/portposdlg.cpp
--- kdemultimedia.orig/arts/builder/portposdlg.cpp	Sat Jun 26 12:12:14 2004
+++ kdemultimedia/arts/builder/portposdlg.cpp	Mon Oct 18 16:17:31 2004
@@ -220,7 +220,7 @@
 
 		if(direction == 0) finddirection = ModulePort::out;
 
-		for(ci = cl.begin(); ci != cl.end(); ci++)
+		for(ci = cl.begin(); ci != cl.end(); ++ci)
 		{
 			StructureComponent *component = *ci;
 			if(component->type() == StructureComponent::ctPort)
diff -urN -x CVS kdemultimedia.orig/arts/builder/propertypanel.cpp kdemultimedia/arts/builder/propertypanel.cpp
--- kdemultimedia.orig/arts/builder/propertypanel.cpp	Thu Jul 10 02:49:54 2003
+++ kdemultimedia/arts/builder/propertypanel.cpp	Mon Oct 18 16:17:31 2004
@@ -377,7 +377,7 @@
 	constantValueComboBox->clear();
 
 	std::vector<Arts::EnumComponent>::const_iterator eci;
-	for(eci = edef.contents.begin(); eci != edef.contents.end(); eci++)
+	for(eci = edef.contents.begin(); eci != edef.contents.end(); ++eci)
 		constantValueComboBox->insertItem(QString::fromUtf8(eci->name.c_str()));
 }
 
diff -urN -x CVS kdemultimedia.orig/arts/builder/retrievedlg.cpp kdemultimedia/arts/builder/retrievedlg.cpp
--- kdemultimedia.orig/arts/builder/retrievedlg.cpp	Sun Apr 11 11:02:39 2004
+++ kdemultimedia/arts/builder/retrievedlg.cpp	Mon Oct 18 16:17:31 2004
@@ -81,7 +81,7 @@
 		names.insert(structures[i]->Name());
 
 	set<string>::iterator ni;
-	for(ni=names.begin();ni!=names.end();ni++)
+	for(ni=names.begin();ni!=names.end();++ni)
 		listbox->insertItem((*ni).c_str());
 #endif
 
diff -urN -x CVS kdemultimedia.orig/arts/builder/structure.cpp kdemultimedia/arts/builder/structure.cpp
--- kdemultimedia.orig/arts/builder/structure.cpp	Sun Jun 22 23:03:19 2003
+++ kdemultimedia/arts/builder/structure.cpp	Mon Oct 18 16:17:31 2004
@@ -215,13 +215,13 @@
 
 	arts_debug("clc");
 /*
-	for(ci = ComponentList.begin(); ci != ComponentList.end(); ci++)
+	for(ci = ComponentList.begin(); ci != ComponentList.end(); ++ci)
 		delete (*ci);
 
 	ComponentList.erase(ComponentList.begin(), ComponentList.end());
 	ModuleList.erase(ModuleList.begin(), ModuleList.end());
 */
-	for(ci = ComponentList.begin(); ci != ComponentList.end(); ci++)
+	for(ci = ComponentList.begin(); ci != ComponentList.end(); ++ci)
 		(*ci)->setSelected(true);
 	deleteSelected();
 
@@ -298,7 +298,7 @@
 	vector<Arts::ModuleDesc> *modules = StructureDesc.modules();
 	vector<Arts::ModuleDesc>::iterator mi;
 
-	for(mi=modules->begin(); mi != modules->end(); mi++)
+	for(mi=modules->begin(); mi != modules->end(); ++mi)
 	{
 		Module *m = new Module(*mi,StructureDesc,canvas);
 
@@ -311,7 +311,7 @@
 	vector<Arts::StructurePortDesc> *ports = StructureDesc.ports();
 	vector<Arts::StructurePortDesc>::iterator pi;
 
-	for(pi=ports->begin(); pi != ports->end(); pi++)
+	for(pi=ports->begin(); pi != ports->end(); ++pi)
 	{
 		StructurePort *p = new StructurePort(*pi,StructureDesc,canvas);
 
@@ -390,7 +390,7 @@
 	list<StructureComponent *>::iterator ci;
 	long targets = 0;
 
-	for(ci=ComponentList.begin();ci!=ComponentList.end();ci++)
+	for(ci=ComponentList.begin();ci!=ComponentList.end();++ci)
 		if((*ci)->selected()) targets++;
 
 	return targets;
@@ -418,12 +418,12 @@
 	list<StructureComponent *>::iterator ci;
 	list<ModulePort *> allPorts;
 
-	for(ci = ComponentList.begin(); ci!=ComponentList.end(); ci++)
+	for(ci = ComponentList.begin(); ci!=ComponentList.end(); ++ci)
 		if((*ci)->selected())
 			(*ci)->dumpPorts(allPorts);
 
 	list<ModulePort *>::iterator pi;
-	for(pi = allPorts.begin(); pi != allPorts.end(); pi++)
+	for(pi = allPorts.begin(); pi != allPorts.end(); ++pi)
 		(*pi)->PortDesc.disconnectAll();
 
 	// and from the ComponentList (the future ;)
@@ -446,7 +446,7 @@
 {
 	list<StructureComponent *>::iterator ci;
 
-	for(ci=ComponentList.begin();ci!=ComponentList.end();ci++)
+	for(ci=ComponentList.begin();ci!=ComponentList.end();++ci)
 	{
 		StructureComponent *c = *ci;
 
diff -urN -x CVS kdemultimedia.orig/juk/k3bexporter.cpp kdemultimedia/juk/k3bexporter.cpp
--- kdemultimedia.orig/juk/k3bexporter.cpp	Mon Jul  5 09:29:29 2004
+++ kdemultimedia/juk/k3bexporter.cpp	Sat Oct 16 20:36:23 2004
@@ -222,8 +222,8 @@
 void K3bPlaylistExporter::slotExport()
 {
     if(m_playlistBox) {
-        setPlaylist(m_playlistBox->currentPlaylist());
-        exportPlaylistItems(m_playlistBox->currentPlaylist()->items());
+        setPlaylist(m_playlistBox->visiblePlaylist());
+        exportPlaylistItems(m_playlistBox->visiblePlaylist()->items());
     }
 }
 
diff -urN -x CVS kdemultimedia.orig/juk/playermanager.cpp kdemultimedia/juk/playermanager.cpp
--- kdemultimedia.orig/juk/playermanager.cpp	Sat Aug 14 07:40:00 2004
+++ kdemultimedia/juk/playermanager.cpp	Tue Oct  5 17:16:57 2004
@@ -469,6 +469,7 @@
     stop();
     delete m_player;
     m_player = createPlayer(system);
+    setup();
 }
 
 void PlayerManager::slotSetVolume(int volume)
diff -urN -x CVS kdemultimedia.orig/juk/playlist.cpp kdemultimedia/juk/playlist.cpp
--- kdemultimedia.orig/juk/playlist.cpp	Wed Sep 15 11:32:20 2004
+++ kdemultimedia/juk/playlist.cpp	Mon Oct 18 16:34:37 2004
@@ -874,7 +874,7 @@
 
     QStringList fileList;
 
-    for(KURL::List::Iterator it = urls.begin(); it != urls.end(); it++)
+    for(KURL::List::Iterator it = urls.begin(); it != urls.end(); ++it)
 	fileList.append((*it).path());
 
     addFiles(fileList, m_collection->importPlaylists(), after);
diff -urN -x CVS kdemultimedia.orig/juk/playlistbox.cpp kdemultimedia/juk/playlistbox.cpp
--- kdemultimedia.orig/juk/playlistbox.cpp	Sat Oct  2 07:10:40 2004
+++ kdemultimedia/juk/playlistbox.cpp	Mon Oct 18 16:34:37 2004
@@ -210,7 +210,9 @@
     if(m_dynamicPlaylist)
 	return m_dynamicPlaylist;
 
-    if(currentItem() && static_cast<Item *>(currentItem())->playlist())
+    if(Playlist::playingItem())
+	return Playlist::playingItem()->playlist();
+    else if(currentItem() && static_cast<Item *>(currentItem())->playlist())
 	return static_cast<Item *>(currentItem())->playlist();
     else
 	return PlaylistCollection::currentPlaylist();
@@ -332,7 +334,7 @@
     if ( m_dynamicPlaylist )
 	return m_dynamicPlaylist;
 
-    return PlaylistCollection::currentPlaylist();
+    return selectedItems().front()->playlist();
 }
 
 void PlaylistBox::slotPlaylistDestroyed(Playlist *p)
@@ -385,7 +387,7 @@
 
     if(KURLDrag::decode(s, urls) && !urls.isEmpty()) {
 	QStringList files;
-	for(KURL::List::Iterator it = urls.begin(); it != urls.end(); it++)
+	for(KURL::List::Iterator it = urls.begin(); it != urls.end(); ++it)
 	    files.append((*it).path());
 
 	if(item) {
@@ -525,11 +527,12 @@
     KListView::keyReleaseEvent(e);
 }
 
-PlaylistBox::ItemList PlaylistBox::selectedItems()
+PlaylistBox::ItemList PlaylistBox::selectedItems() const
 {
     ItemList l;
 
-    for(QListViewItemIterator it(this, QListViewItemIterator::Selected); it.current(); ++it)
+    for(QListViewItemIterator it(const_cast<PlaylistBox *>(this),
+				 QListViewItemIterator::Selected); it.current(); ++it)
 	l.append(static_cast<Item *>(*it));
 
     return l;
diff -urN -x CVS kdemultimedia.orig/juk/playlistbox.h kdemultimedia/juk/playlistbox.h
--- kdemultimedia.orig/juk/playlistbox.h	Thu Sep 30 10:11:46 2004
+++ kdemultimedia/juk/playlistbox.h	Sat Oct 16 20:36:23 2004
@@ -89,7 +89,7 @@
     virtual void keyPressEvent(QKeyEvent *e);
     virtual void keyReleaseEvent(QKeyEvent *e);
 
-    QValueList<Item *> selectedItems();
+    QValueList<Item *> selectedItems() const;
     void setSingleItem(QListViewItem *item);
 
     void setupItem(Item *item);
diff -urN -x CVS kdemultimedia.orig/juk/playlistsplitter.cpp kdemultimedia/juk/playlistsplitter.cpp
--- kdemultimedia.orig/juk/playlistsplitter.cpp	Thu Sep 30 10:11:46 2004
+++ kdemultimedia/juk/playlistsplitter.cpp	Sat Oct 16 20:36:23 2004
@@ -97,8 +97,8 @@
 
     connect(m_playlistBox->object(), SIGNAL(signalSelectedItemsChanged()),
             this, SLOT(slotPlaylistSelectionChanged()));
-    connect(m_playlistBox, SIGNAL(signalPlaylistDestroyed(Playlist*)),
-            m_editor, SLOT(slotPlaylistDestroyed(Playlist*)));
+    connect(m_playlistBox, SIGNAL(signalPlaylistDestroyed(Playlist *)),
+            m_editor, SLOT(slotPlaylistDestroyed(Playlist *)));
 
     moveToFirst(m_playlistBox);
 
@@ -157,7 +157,7 @@
 
 void PlaylistSplitter::slotPlaylistSelectionChanged()
 {
-    m_editor->slotSetItems(static_cast<PlaylistCollection *>(m_playlistBox)->selectedItems());
+    m_editor->slotSetItems(visiblePlaylist()->selectedItems());
 }
 
 void PlaylistSplitter::slotPlaylistChanged(QWidget *w)
diff -urN -x CVS kdemultimedia.orig/juk/statuslabel.cpp kdemultimedia/juk/statuslabel.cpp
--- kdemultimedia.orig/juk/statuslabel.cpp	Sat Oct  2 07:10:40 2004
+++ kdemultimedia/juk/statuslabel.cpp	Sat Oct 16 20:36:23 2004
@@ -28,8 +28,6 @@
 #include "playlistinterface.h"
 #include "actioncollection.h"
 #include "tag.h"
-#include "playlist.h"
-#include "playlistitem.h"
 
 using namespace ActionCollection;
 
@@ -101,16 +99,8 @@
 
 void StatusLabel::updateCurrent()
 {
-    //kdDebug(65432) << k_funcinfo << endl;
-
-    // Use a hack to always show the currently playing playlist in the status
-    // bar.  We can't change PlaylistBox::currentPlaylist() to do this as that
-    // has too many sideeffects at this point.
-
-    if(Playlist::playingItem()) {
-        PlaylistInterface *playlist = Playlist::playingItem()->playlist();
-
-        FileHandle file = playlist->currentFile();
+    if(playlist()->playing()) {
+        FileHandle file = playlist()->currentFile();
 
         QString mid =  file.tag()->artist().isEmpty() || file.tag()->title().isEmpty()
             ? QString::null : QString(" - ");
@@ -118,15 +108,13 @@
         QString text = file.tag()->artist() + mid + file.tag()->title();
 
         m_trackLabel->setText(text);
-        m_playlistLabel->setText(playlist->name().simplifyWhiteSpace());
+        m_playlistLabel->setText(playlist()->name().simplifyWhiteSpace());
     }
-    else
-        updateData();
 }
 
 void StatusLabel::updateData()
 {
-    //kdDebug(65432) << k_funcinfo << endl;
+    updateCurrent();
 
     if(!playlist()->playing()) {
         setItemTotalTime(0);
@@ -151,8 +139,6 @@
         m_playlistLabel->setText(playlist()->name());
         m_trackLabel->setText(i18n("1 item", "%n items", playlist()->count()) + " - " + time);
     }
-    else
-        updateCurrent();
 }
 
 ////////////////////////////////////////////////////////////////////////////////
diff -urN -x CVS kdemultimedia.orig/juk/tageditor.cpp kdemultimedia/juk/tageditor.cpp
--- kdemultimedia.orig/juk/tageditor.cpp	Mon Aug 23 05:17:28 2004
+++ kdemultimedia/juk/tageditor.cpp	Mon Oct 18 16:34:37 2004
@@ -156,6 +156,15 @@
     if(m_performingSave)
 	return;
 
+    // Store the playlist that we're setting because saveChangesPrompt
+    // can delete the PlaylistItems in list.
+
+    Playlist *itemPlaylist = 0;
+    if(!list.isEmpty())
+	itemPlaylist = list.first()->playlist();
+
+    bool hadPlaylist = m_currentPlaylist != 0;
+
     saveChangesPrompt();
 
     if(m_currentPlaylist) {
@@ -163,7 +172,17 @@
 		   this, SLOT(slotItemRemoved(PlaylistItem *)));
     }
 
-    m_currentPlaylist = list.isEmpty() ? 0 : static_cast<Playlist *>(list.first()->listView());
+    if(hadPlaylist && !m_currentPlaylist || !itemPlaylist) {
+	m_currentPlaylist = 0;
+	m_items.clear();
+    }
+    else {
+	m_currentPlaylist = itemPlaylist;
+
+	// We can't use list here, it may not be valid
+
+	m_items = itemPlaylist->selectedItems();
+    }
 
     if(m_currentPlaylist) {
 	connect(m_currentPlaylist, SIGNAL(signalAboutToRemove(PlaylistItem *)),
@@ -171,8 +190,6 @@
 	connect(m_currentPlaylist, SIGNAL(destroyed()), this, SLOT(slotPlaylistRemoved()));
     }
 
-    m_items = list;
-
     if(isVisible())
 	slotRefresh();
     else
@@ -701,7 +718,7 @@
 
     QStringList files;
 
-    for(PlaylistItemList::Iterator it = m_items.begin(); it != m_items.end(); it++)
+    for(PlaylistItemList::Iterator it = m_items.begin(); it != m_items.end(); ++it)
 	files.append((*it)->file().absFilePath());
 
     if(KMessageBox::questionYesNoList(this,
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/ams.desktop kdemultimedia/kappfinder-data/ams.desktop
--- kdemultimedia.orig/kappfinder-data/ams.desktop	Fri Sep 24 15:46:43 2004
+++ kdemultimedia/kappfinder-data/ams.desktop	Tue Oct  5 15:20:19 2004
@@ -22,9 +22,10 @@
 GenericName[pt]=Sintetizador Modular do Alsa
 GenericName[pt_BR]=Sintetizador Modular do Alsa
 GenericName[ru]=Модульный синтезатор ALSA
+GenericName[sk]=Modulárny syntezátor pre Alsa
 GenericName[sl]=Modularni sintetizator Alsa
 GenericName[sr]=Модуларни Alsa-ин синтетизатор
-GenericName[sr@Latn]=Модуларни Alsa-ин синтетизатор
+GenericName[sr@Latn]=Modularni Alsa-in sintetizator
 GenericName[sv]=Alsa modulär synthesizer
 GenericName[ta]=அல்சா ஏற்ற இறக்க தொகுப்பான்
 GenericName[tg]=Ҳамзамонсози Alsa Modular
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/amsynth.desktop kdemultimedia/kappfinder-data/amsynth.desktop
--- kdemultimedia.orig/kappfinder-data/amsynth.desktop	Sat Sep 25 15:40:46 2004
+++ kdemultimedia/kappfinder-data/amsynth.desktop	Tue Oct  5 15:20:19 2004
@@ -23,8 +23,9 @@
 GenericName[pt]=Retro Analógico - Sintetizador por Software
 GenericName[pt_BR]=Retro Analog - sintetização de software de modelagem
 GenericName[ru]=Ретро - программный синтезатор
+GenericName[sk]=Retro-analógované modelovanie Softsynth
 GenericName[sr]=Ретро аналогно моделирање софтверског синтетизатора
-GenericName[sr@Latn]=Ретро аналогно моделирање софтверског синтетизатора
+GenericName[sr@Latn]=Retro analogno modeliranje softverskog sintetizatora
 GenericName[sv]=Programvarusynt med gammaldags analogmodellering
 GenericName[ta]=ரெட்ரோ அனலாக் -  மாடலிங் சாஃப்ட்சின்த்
 GenericName[tg]=Ҷӯркунии Шабеҳ - Моделсозии Ҳамзамонсози Нармафзор
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/ardour.desktop kdemultimedia/kappfinder-data/ardour.desktop
--- kdemultimedia.orig/kappfinder-data/ardour.desktop	Thu Sep 16 15:49:10 2004
+++ kdemultimedia/kappfinder-data/ardour.desktop	Tue Oct  5 15:20:19 2004
@@ -24,8 +24,9 @@
 GenericName[pt]=Estúdio Áudio Multi-faixa
 GenericName[pt_BR]=Estúdio de Áudio Multitrack 
 GenericName[ru]=Аудиостудия
+GenericName[sk]=Audio studio pre niekoľko stôp
 GenericName[sr]=Вишетрачни аудио студио
-GenericName[sr@Latn]=Вишетрачни аудио студио
+GenericName[sr@Latn]=Višetračni audio studio
 GenericName[sv]=Flerspårs ljudstudio
 GenericName[ta]=பலவிதபாதையடைய கேட்பொலி ஸ்டுடியோ
 GenericName[tg]=Студияи Садоии Бисёршиора
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/djplay.desktop kdemultimedia/kappfinder-data/djplay.desktop
--- kdemultimedia.orig/kappfinder-data/djplay.desktop	Thu Sep 16 15:49:10 2004
+++ kdemultimedia/kappfinder-data/djplay.desktop	Tue Oct  5 15:20:19 2004
@@ -29,9 +29,10 @@
 GenericName[pt]=Leitor e Mistura para DJs
 GenericName[pt_BR]=Reprodutor e Mixer
 GenericName[ru]=DJ-Mixer и проигрыватель
+GenericName[sk]=DJ-Mixer a prehrávač
 GenericName[sl]=DJ-mešalnik in predvajalnik
 GenericName[sr]=DJ-Mixer и плејер
-GenericName[sr@Latn]=DJ-Mixer и плејер
+GenericName[sr@Latn]=DJ-Mixer i plejer
 GenericName[sv]=DJ-mixer och ljudspelare
 GenericName[ta]=டிஜே-ஒன்றுசேர்த்து இயக்கும்
 GenericName[tg]=DJ-Омехтакунак ва Бозингар
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/ecamegapedal.desktop kdemultimedia/kappfinder-data/ecamegapedal.desktop
--- kdemultimedia.orig/kappfinder-data/ecamegapedal.desktop	Sun Sep 12 15:56:24 2004
+++ kdemultimedia/kappfinder-data/ecamegapedal.desktop	Mon Oct  4 15:25:25 2004
@@ -13,7 +13,7 @@
 GenericName[nn]=Ecasound-effektspor
 GenericName[pt_BR]=Faixas de Efeito do Ecasound
 GenericName[sr]=Ecasound трака са ефектима
-GenericName[sr@Latn]=Ecasound трака са ефектима
+GenericName[sr@Latn]=Ecasound traka sa efektima
 GenericName[sv]=Ecasound-effektspår
 GenericName[ta]=எகாஒலி எபெஸ்பாதை
 StartupNotify=true
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/freebirth.desktop kdemultimedia/kappfinder-data/freebirth.desktop
--- kdemultimedia.orig/kappfinder-data/freebirth.desktop	Thu Sep 16 15:49:10 2004
+++ kdemultimedia/kappfinder-data/freebirth.desktop	Tue Oct  5 15:20:19 2004
@@ -25,8 +25,9 @@
 GenericName[pl]=Maszyna perkusyjna
 GenericName[pt]=Máquina de Percussão
 GenericName[pt_BR]=Bateria
+GenericName[sk]=Nástroj pre bicie
 GenericName[sr]=Бубањ машина
-GenericName[sr@Latn]=Бубањ машина
+GenericName[sr@Latn]=Bubanj mašina
 GenericName[sv]=Trummaskin
 GenericName[ta]=டிரம் இயந்திரம்
 GenericName[tg]=Мошинаи Таблак
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/freqtweak.desktop kdemultimedia/kappfinder-data/freqtweak.desktop
--- kdemultimedia.orig/kappfinder-data/freqtweak.desktop	Thu Sep 16 15:49:10 2004
+++ kdemultimedia/kappfinder-data/freqtweak.desktop	Tue Oct  5 15:20:19 2004
@@ -27,9 +27,10 @@
 GenericName[pt]=Efeitos para o Jack
 GenericName[pt_BR]=Efeitos para o Jack
 GenericName[ru]=Эффекты для Jack
+GenericName[sk]=Efekty pre Jack
 GenericName[sl]=Učinki za Jack
 GenericName[sr]=Ефекти за Jack
-GenericName[sr@Latn]=Ефекти за Jack
+GenericName[sr@Latn]=Efekti za Jack
 GenericName[sv]=Effekter för Jack
 GenericName[ta]=ஜாக்குக்கான ஒலி அமைப்புகள
 GenericName[tg]=Натиҷаҳо барои Jack
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/galan.desktop kdemultimedia/kappfinder-data/galan.desktop
--- kdemultimedia.orig/kappfinder-data/galan.desktop	Thu Sep 16 15:49:10 2004
+++ kdemultimedia/kappfinder-data/galan.desktop	Tue Oct  5 15:20:19 2004
@@ -24,8 +24,9 @@
 GenericName[pt]=Sintetizador Modular
 GenericName[pt_BR]=Sintetizador Modular
 GenericName[ru]=Синтезатор
+GenericName[sk]=Modulárny syntezátor
 GenericName[sr]=Модуларни синтетизатор
-GenericName[sr@Latn]=Модуларни синтетизатор
+GenericName[sr@Latn]=Modularni sintetizator
 GenericName[sv]=Modulär synthesizer
 GenericName[ta]= மாடுலர் சிந்த்
 GenericName[tg]=Ҳамзамонсози Модулӣ
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/hydrogen.desktop kdemultimedia/kappfinder-data/hydrogen.desktop
--- kdemultimedia.orig/kappfinder-data/hydrogen.desktop	Thu Sep 16 15:49:10 2004
+++ kdemultimedia/kappfinder-data/hydrogen.desktop	Tue Oct  5 15:20:19 2004
@@ -27,8 +27,9 @@
 GenericName[pl]=Maszyna perkusyjna
 GenericName[pt]=Máquina de Percussão
 GenericName[pt_BR]=Bateria
+GenericName[sk]=Nástroj pre bicie
 GenericName[sr]=Бубањ машина
-GenericName[sr@Latn]=Бубањ машина
+GenericName[sr@Latn]=Bubanj mašina
 GenericName[sv]=Trummaskin
 GenericName[ta]=டிரம் இயந்திரம்
 GenericName[tg]=Мошинаи Таблак
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/jack-rack.desktop kdemultimedia/kappfinder-data/jack-rack.desktop
--- kdemultimedia.orig/kappfinder-data/jack-rack.desktop	Thu Sep 16 15:49:10 2004
+++ kdemultimedia/kappfinder-data/jack-rack.desktop	Mon Oct  4 15:25:25 2004
@@ -22,7 +22,7 @@
 GenericName[pt]=Efeitos Jack
 GenericName[pt_BR]=Faixas de Efeito do Jack
 GenericName[sr]=Jack-ова трака са ефектима
-GenericName[sr@Latn]=Jack-ова трака са ефектима
+GenericName[sr@Latn]=Jack-ova traka sa efektima
 GenericName[sv]=Effektbord för Jack
 GenericName[ta]=ஜாக் ஒலி அமைப்பு பாதை
 StartupNotify=true
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/jamin.desktop kdemultimedia/kappfinder-data/jamin.desktop
--- kdemultimedia.orig/kappfinder-data/jamin.desktop	Fri Sep 24 15:46:43 2004
+++ kdemultimedia/kappfinder-data/jamin.desktop	Tue Oct  5 15:20:19 2004
@@ -27,8 +27,9 @@
 GenericName[pt]=Ferramenta de masterização do Jack
 GenericName[pt_BR]=Ferramenta de masterização Jack
 GenericName[ru]=Утилита мастеринга Jack
+GenericName[sk]=Nahrávací nástroj pre Jack
 GenericName[sr]=Jack-ов алат за управљање
-GenericName[sr@Latn]=Jack-ов алат за управљање
+GenericName[sr@Latn]=Jack-ov alat za upravljanje
 GenericName[sv]=Jack-inspelningsverktyg
 GenericName[ta]=ஜாக் முதன்மை கருவி
 GenericName[tg]=Асбоби устодии Jack
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/meterbridge.desktop kdemultimedia/kappfinder-data/meterbridge.desktop
--- kdemultimedia.orig/kappfinder-data/meterbridge.desktop	Thu Sep 16 15:49:10 2004
+++ kdemultimedia/kappfinder-data/meterbridge.desktop	Tue Oct  5 15:20:19 2004
@@ -27,9 +27,10 @@
 GenericName[pt]=Medidor VU de Dois Canais
 GenericName[pt_BR]=Vu-Meter de Canal duplo
 GenericName[ru]=Двухканальный измеритель уровня
+GenericName[sk]=VU-Meter pre dvojitý kanál
 GenericName[sl]=Dvokanalni prikazovalnik glasnosti
 GenericName[sr]=Двоканални VU метар
-GenericName[sr@Latn]=Двоканални VU метар
+GenericName[sr@Latn]=Dvokanalni VU metar
 GenericName[sv]=VU-mätare med två kanaler
 GenericName[ta]=இரட்டை அலை இடை வெளிப்பகுதி VU-அளவுக்கருவி
 GenericName[tg]=Маҷрои Дучанди VU-Meter
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/mixxx.desktop kdemultimedia/kappfinder-data/mixxx.desktop
--- kdemultimedia.orig/kappfinder-data/mixxx.desktop	Thu Sep 16 15:49:10 2004
+++ kdemultimedia/kappfinder-data/mixxx.desktop	Tue Oct  5 15:20:19 2004
@@ -27,9 +27,10 @@
 GenericName[pt]=Leitor e Mistura para DJs
 GenericName[pt_BR]=Reprodutor e Mixer
 GenericName[ru]=DJ-Mixer и проигрыватель
+GenericName[sk]=DJ-Mixer a prehrávač
 GenericName[sl]=DJ-mešalnik in predvajalnik
 GenericName[sr]=DJ-Mixer и плејер
-GenericName[sr@Latn]=DJ-Mixer и плејер
+GenericName[sr@Latn]=DJ-Mixer i plejer
 GenericName[sv]=DJ-mixer och ljudspelare
 GenericName[ta]=டிஜே-ஒன்றுசேர்த்து இயக்கும்
 GenericName[tg]=DJ-Омехтакунак ва Бозингар
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/muse.desktop kdemultimedia/kappfinder-data/muse.desktop
--- kdemultimedia.orig/kappfinder-data/muse.desktop	Thu Sep 16 15:49:10 2004
+++ kdemultimedia/kappfinder-data/muse.desktop	Tue Oct  5 15:20:19 2004
@@ -26,9 +26,10 @@
 GenericName[pt]=Sequenciador de Música
 GenericName[pt_BR]=Seqüenciador de Músicas
 GenericName[ru]=Звуковой сиквенсор
+GenericName[sk]=Sekvencér pre hudbu
 GenericName[sl]=Glasbeni sekvenčnik
 GenericName[sr]=Музички секвенцер
-GenericName[sr@Latn]=Музички секвенцер
+GenericName[sr@Latn]=Muzički sekvencer
 GenericName[sv]=Musik-sequencer
 GenericName[ta]=இசை வரிசைபடுத்தி
 GenericName[tg]=Тартибгузори Мусиқӣ
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/qjackctl.desktop kdemultimedia/kappfinder-data/qjackctl.desktop
--- kdemultimedia.orig/kappfinder-data/qjackctl.desktop	Thu Sep 16 15:49:10 2004
+++ kdemultimedia/kappfinder-data/qjackctl.desktop	Tue Oct  5 15:20:19 2004
@@ -28,9 +28,10 @@
 GenericName[pt]=Controlo do Jack
 GenericName[pt_BR]=Controle para o Jack
 GenericName[ru]=Управление Jack
+GenericName[sk]=Ovládanie pre Jack
 GenericName[sl]=Nadzor za Jack
 GenericName[sr]=Контрола за Jack
-GenericName[sr@Latn]=Контрола за Jack
+GenericName[sr@Latn]=Kontrola za Jack
 GenericName[sv]=Styrning av Jack
 GenericName[ta]=ஜாக்குக்கான கட்டுப்பாடு
 GenericName[tg]=Идора барои Jack
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/qsynth.desktop kdemultimedia/kappfinder-data/qsynth.desktop
--- kdemultimedia.orig/kappfinder-data/qsynth.desktop	Thu Sep 16 15:49:10 2004
+++ kdemultimedia/kappfinder-data/qsynth.desktop	Tue Oct  5 15:20:19 2004
@@ -29,9 +29,10 @@
 GenericName[pt]=Controlo do FluidSynth
 GenericName[pt_BR]=Controle para o FluidSynth
 GenericName[ru]=Управление FluidSynth
+GenericName[sk]=Ovládanie pre FluidSynth
 GenericName[sl]=Nadzor za FluidSynth
 GenericName[sr]=Контрола за FluidSynth
-GenericName[sr@Latn]=Контрола за FluidSynth
+GenericName[sr@Latn]=Kontrola za FluidSynth
 GenericName[sv]=Styrning av Fluidsynth
 GenericName[ta]=ப்ளூயிட்சின்த்தின் கட்டுப்பாடு
 GenericName[tg]=Идора барои FluidSynth
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/rezound.desktop kdemultimedia/kappfinder-data/rezound.desktop
--- kdemultimedia.orig/kappfinder-data/rezound.desktop	Thu Sep 16 15:49:10 2004
+++ kdemultimedia/kappfinder-data/rezound.desktop	Tue Oct  5 15:20:19 2004
@@ -31,9 +31,10 @@
 GenericName[pt]=Editor de Áudio
 GenericName[pt_BR]=Editor de Áudio
 GenericName[ru]=Звуковой редактор
+GenericName[sk]=Audio editor
 GenericName[sl]=Urejevalnik zvoka
 GenericName[sr]=Аудио уређивач
-GenericName[sr@Latn]=Аудио уређивач
+GenericName[sr@Latn]=Audio uređivač
 GenericName[sv]=Ljudeditor
 GenericName[ta]=கேட்பொலி தொகுப்பாளர்
 GenericName[tg]=Муҳаррири Садо
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/vkeybd.desktop kdemultimedia/kappfinder-data/vkeybd.desktop
--- kdemultimedia.orig/kappfinder-data/vkeybd.desktop	Thu Sep 16 15:49:10 2004
+++ kdemultimedia/kappfinder-data/vkeybd.desktop	Tue Oct  5 15:20:19 2004
@@ -28,9 +28,10 @@
 GenericName[pt]=Teclado MIDI Virtual
 GenericName[pt_BR]=Teclado MIDI Virtual
 GenericName[ru]=Виртуальная клавиатура MIDI
+GenericName[sk]=Virtuálne klávesy MIDI
 GenericName[sl]=Navidezna klaviatura MIDI
 GenericName[sr]=Виртуелна MIDI клавијатура
-GenericName[sr@Latn]=Виртуелна MIDI клавијатура
+GenericName[sr@Latn]=Virtuelna MIDI klavijatura
 GenericName[sv]=Virtuellt MIDI-keyboard
 GenericName[ta]=மாய மேசை மிடி விசைப்பலகை
 GenericName[tg]=Забонаки Маҷозии MIDI
diff -urN -x CVS kdemultimedia.orig/kappfinder-data/zynaddsubfx.desktop kdemultimedia/kappfinder-data/zynaddsubfx.desktop
--- kdemultimedia.orig/kappfinder-data/zynaddsubfx.desktop	Sat Sep 25 15:40:46 2004
+++ kdemultimedia/kappfinder-data/zynaddsubfx.desktop	Mon Oct  4 15:25:25 2004
@@ -26,7 +26,7 @@
 GenericName[pt_BR]=Sintetizador de software
 GenericName[ru]=Программный синтезатор
 GenericName[sr]=Софтверски синтетизатор
-GenericName[sr@Latn]=Софтверски синтетизатор
+GenericName[sr@Latn]=Softverski sintetizator
 GenericName[sv]=Programvarusynt
 GenericName[ta]=மென்மையான சிந்த்
 GenericName[tg]=Ҳамзамонсози Нармафзор
diff -urN -x CVS kdemultimedia.orig/kaudiocreator/eventsrc kdemultimedia/kaudiocreator/eventsrc
--- kdemultimedia.orig/kaudiocreator/eventsrc	Sat Sep 25 15:40:48 2004
+++ kdemultimedia/kaudiocreator/eventsrc	Tue Oct  5 15:20:21 2004
@@ -31,7 +31,7 @@
 Name[ru]=Дорожка считана
 Name[sl]=Skladba zajeta
 Name[sr]=Нумера је изгребована
-Name[sr@Latn]=Нумера је изгребована
+Name[sr@Latn]=Numera je izgrebovana
 Name[sv]=Spår lagrat
 Name[ta]=தடம் பிரிக்கப்பட்டது
 Name[tg]=Шиор Канда шудааст
@@ -60,9 +60,10 @@
 Comment[pt]=A faixa foi extraída
 Comment[pt_BR]=Finalização da conversão da faixa
 Comment[ru]=Считывание дорожки завершено.
+Comment[sk]=Dokončené ripovanie stopy
 Comment[sl]=Zajemanje skladbe končano
 Comment[sr]=Гребовање нумере је завршено
-Comment[sr@Latn]=Гребовање нумере је завршено
+Comment[sr@Latn]=Grebovanje numere je završeno
 Comment[sv]=Spåret har lagrats klart
 Comment[ta]=தடம் பிரித்தலை முடித்தது
 Comment[tg]=Шиор кандашавиро ба охир расонд
@@ -92,9 +93,10 @@
 Name[pt]=CD Extraído
 Name[pt_BR]=CD Convertido
 Name[ru]=CD считан
+Name[sk]=Ripovanie CD dokončené
 Name[sl]=CD zajet
 Name[sr]=CD је изгребован
-Name[sr@Latn]=CD је изгребован
+Name[sr@Latn]=CD je izgrebovan
 Name[sv]=Cd lagrad
 Name[ta]=CD பிரிக்கப்பட்டது
 Name[tg]=Диски Фишурда Канда шудааст
@@ -123,9 +125,10 @@
 Comment[pt]=O CD foi extraído
 Comment[pt_BR]=Finalização da conversão do CD
 Comment[ru]=Считывание CD завершено.
+Comment[sk]=Dokončené ripovanie CD
 Comment[sl]=Zajemanje CD-ja končano
 Comment[sr]=Гребовање CD-а је завршено
-Comment[sr@Latn]=Гребовање CD-а је завршено
+Comment[sr@Latn]=Grebovanje CD-a je završeno
 Comment[sv]=Cd:n har lagrats klart
 Comment[ta]=CD பிரித்தல் முடிந்தது
 Comment[tg]=Диски Фишурда кандашавиро ба охир расонд
@@ -156,9 +159,10 @@
 Name[pt]=Faixa Codificada
 Name[pt_BR]=Faixa codificada
 Name[ru]=Дорожка преобразована
+Name[sk]=Ripovanie stopy dokončené
 Name[sl]=Skladba kodirana
 Name[sr]=Нумера је искодирана
-Name[sr@Latn]=Нумера је искодирана
+Name[sr@Latn]=Numera je iskodirana
 Name[sv]=Spår kodat
 Name[ta]=தடம் மறையாக்கப்பட்டது
 Name[tg]=Шиор Рамзӣ кунонида шудааст
@@ -187,9 +191,10 @@
 Comment[pt]=A faixa foi codificada
 Comment[pt_BR]=Finalização da codificação da faixa
 Comment[ru]=Преобразование дорожки завершено
+Comment[sk]=Dokončené kódovanie stopy
 Comment[sl]=Kodiranje skladbe končano
 Comment[sr]=Кодирање нумере је завршено
-Comment[sr@Latn]=Кодирање нумере је завршено
+Comment[sr@Latn]=Kodiranje numere je završeno
 Comment[sv]=Spåret har kodats klart
 Comment[ta]=தடம் மறையாக்கத்தை முடித்தது
 Comment[tg]=Шиор рамзикунониро ба охир расонд
@@ -220,9 +225,10 @@
 Name[pt]=CD Codificado
 Name[pt_BR]=CD Codificado
 Name[ru]=Преобразование CD завершено
+Name[sk]=CD zakódované
 Name[sl]=CD kodiran
 Name[sr]=CD је искодиран
-Name[sr@Latn]=CD је искодиран
+Name[sr@Latn]=CD je iskodiran
 Name[sv]=Cd kodad
 Name[ta]=CD மறையாக்கப்பட்டது
 Name[tg]=Диски Фишурда Рамзӣ кунонида шудааст
@@ -251,9 +257,10 @@
 Comment[pt]=O CD foi codificado
 Comment[pt_BR]=Finalização da codificação do CD
 Comment[ru]=Преобразование диска завершено
+Comment[sk]=Dokončené kódovanie CD
 Comment[sl]=Kodiranje CD-ja končano
 Comment[sr]=Кодирање CD-а је завршено
-Comment[sr@Latn]=Кодирање CD-а је завршено
+Comment[sr@Latn]=Kodiranje CD-a je završeno
 Comment[sv]=Cd:n har kodats klart
 Comment[ta]=CD மறையாக்கத்தை முடித்தது.
 Comment[tg]=Диски Фишурда рамзикунониро ба охир расонд
diff -urN -x CVS kdemultimedia.orig/kfile-plugins/mpc/kfile_mpc.desktop kdemultimedia/kfile-plugins/mpc/kfile_mpc.desktop
--- kdemultimedia.orig/kfile-plugins/mpc/kfile_mpc.desktop	Sat Sep 25 15:40:50 2004
+++ kdemultimedia/kfile-plugins/mpc/kfile_mpc.desktop	Mon Oct  4 15:25:29 2004
@@ -23,7 +23,7 @@
 Name[ru]=Сведения о Musepack
 Name[sl]=Informacije o Musepack
 Name[sr]=Информације о Musepack-у
-Name[sr@Latn]=Информације о Musepack-у
+Name[sr@Latn]=Informacije o Musepack-u
 Name[sv]=Musepack-information
 Name[ta]=MP3 தகவல்
 Name[tg]=Ахбороти Musepack
diff -urN -x CVS kdemultimedia.orig/kfile-plugins/sid/kfile_sid.desktop kdemultimedia/kfile-plugins/sid/kfile_sid.desktop
--- kdemultimedia.orig/kfile-plugins/sid/kfile_sid.desktop	Thu Sep 16 15:49:14 2004
+++ kdemultimedia/kfile-plugins/sid/kfile_sid.desktop	Tue Oct  5 15:20:23 2004
@@ -25,9 +25,10 @@
 Name[pt]=Informação do SID
 Name[pt_BR]=Informação sobre SID
 Name[ru]=Сведения о SID
+Name[sk]=SID info
 Name[sl]=Informacije o SID
 Name[sr]=SID информације
-Name[sr@Latn]=SID информације
+Name[sr@Latn]=SID informacije
 Name[sv]=SID-information
 Name[ta]=SID தகவல்
 Name[tg]=SID Ахборот
diff -urN -x CVS kdemultimedia.orig/kioslave/audiocd/kcmaudiocd/kcmaudiocd.cpp kdemultimedia/kioslave/audiocd/kcmaudiocd/kcmaudiocd.cpp
--- kdemultimedia.orig/kioslave/audiocd/kcmaudiocd/kcmaudiocd.cpp	Sat Sep 18 23:47:29 2004
+++ kdemultimedia/kioslave/audiocd/kcmaudiocd/kcmaudiocd.cpp	Mon Oct  4 00:56:00 2004
@@ -58,7 +58,7 @@
   
     KConfigDialogManager *widget;
     for ( widget = encoderSettings.first(); widget; widget = encoderSettings.next() ){
-      connect(widget, SIGNAL(widgetModified()), this, SLOT(slotConfigChanged()));
+      connect(widget, SIGNAL(widgetModified()), this, SLOT(slotModuleChanged()));
     }
    
     //CDDA Options
@@ -141,6 +141,16 @@
   }
 }
 
+void KAudiocdModule::slotModuleChanged() {
+	KConfigDialogManager *widget;
+	for ( widget = encoderSettings.first(); widget; widget = encoderSettings.next() ){
+		if(widget->hasChanged()){
+			slotConfigChanged();
+			break;
+		}
+	}
+}
+
 void KAudiocdModule::slotConfigChanged() {
 
   if (!configChanged) configChanged = true;
diff -urN -x CVS kdemultimedia.orig/kioslave/audiocd/kcmaudiocd/kcmaudiocd.h kdemultimedia/kioslave/audiocd/kcmaudiocd/kcmaudiocd.h
--- kdemultimedia.orig/kioslave/audiocd/kcmaudiocd/kcmaudiocd.h	Sun Mar 21 04:58:59 2004
+++ kdemultimedia/kioslave/audiocd/kcmaudiocd/kcmaudiocd.h	Mon Oct  4 00:56:00 2004
@@ -51,13 +51,12 @@
 
   void slotConfigChanged();
   void slotEcEnable();
-private:
+  void slotModuleChanged();
 
+private:
   KConfig *config;
   bool configChanged;
 
-  int getBitrateIndex(int value);
-
   QPtrList<KConfigDialogManager> encoderSettings;
 };
 
diff -urN -x CVS kdemultimedia.orig/kioslave/audiocd/plugins/lame/Makefile.am kdemultimedia/kioslave/audiocd/plugins/lame/Makefile.am
--- kdemultimedia.orig/kioslave/audiocd/plugins/lame/Makefile.am	Mon Jul 26 15:22:26 2004
+++ kdemultimedia/kioslave/audiocd/plugins/lame/Makefile.am	Thu Oct  7 14:16:43 2004
@@ -13,3 +13,6 @@
 pluginsdir = $(kde_datadir)/audiocd/plugins
 
 METASOURCES = AUTO
+
+messages: rc.cpp
+	$(XGETTEXT) *.cpp -o $(podir)/audiocd_encoder_lame.pot
diff -urN -x CVS kdemultimedia.orig/kioslave/audiocd/plugins/lame/audiocd_lame_encoder.kcfg kdemultimedia/kioslave/audiocd/plugins/lame/audiocd_lame_encoder.kcfg
--- kdemultimedia.orig/kioslave/audiocd/plugins/lame/audiocd_lame_encoder.kcfg	Sun Mar 21 04:58:59 2004
+++ kdemultimedia/kioslave/audiocd/plugins/lame/audiocd_lame_encoder.kcfg	Mon Oct  4 00:26:56 2004
@@ -20,7 +20,7 @@
     </entry>
     <entry name="quality" type="Int">
       <label>Quality</label>
-      <default>2</default>
+      <default>-2</default>
     </entry>
 
 <!-- Options  -->
diff -urN -x CVS kdemultimedia.orig/kioslave/audiocd/plugins/lame/encoderlame.cpp kdemultimedia/kioslave/audiocd/plugins/lame/encoderlame.cpp
--- kdemultimedia.orig/kioslave/audiocd/plugins/lame/encoderlame.cpp	Thu Sep 16 03:37:07 2004
+++ kdemultimedia/kioslave/audiocd/plugins/lame/encoderlame.cpp	Mon Oct  4 00:26:56 2004
@@ -409,7 +409,7 @@
   Settings *settings = Settings::self();
 
   int quality = settings->quality();
-  if (quality < 0 ) quality = 0;
+  if (quality < 0 ) quality = quality *-1;
   if (quality > 9) quality = 9;
 
   int method = settings->bitrate_constant() ? 0 : 1 ;
diff -urN -x CVS kdemultimedia.orig/kioslave/audiocd/plugins/lame/encoderlameconfig.ui kdemultimedia/kioslave/audiocd/plugins/lame/encoderlameconfig.ui
--- kdemultimedia.orig/kioslave/audiocd/plugins/lame/encoderlameconfig.ui	Sun Jun 20 00:06:55 2004
+++ kdemultimedia/kioslave/audiocd/plugins/lame/encoderlameconfig.ui	Mon Oct  4 00:26:56 2004
@@ -135,7 +135,7 @@
                                 <cstring>TextLabel3_2</cstring>
                             </property>
                             <property name="text">
-                                <string>High</string>
+                                <string>Low</string>
                             </property>
                         </widget>
                         <widget class="QSlider">
@@ -143,16 +143,16 @@
                                 <cstring>kcfg_quality</cstring>
                             </property>
                             <property name="minValue">
-                                <number>0</number>
+                                <number>-9</number>
                             </property>
                             <property name="maxValue">
-                                <number>9</number>
+                                <number>0</number>
                             </property>
                             <property name="pageStep">
                                 <number>1</number>
                             </property>
                             <property name="value">
-                                <number>2</number>
+                                <number>-7</number>
                             </property>
                             <property name="orientation">
                                 <enum>Horizontal</enum>
@@ -166,7 +166,7 @@
                                 <cstring>TextLabel2_2</cstring>
                             </property>
                             <property name="text">
-                                <string>Low</string>
+                                <string>High</string>
                             </property>
                         </widget>
                     </hbox>
diff -urN -x CVS kdemultimedia.orig/kioslave/audiocd/plugins/vorbis/Makefile.am kdemultimedia/kioslave/audiocd/plugins/vorbis/Makefile.am
--- kdemultimedia.orig/kioslave/audiocd/plugins/vorbis/Makefile.am	Mon Jul 26 15:22:26 2004
+++ kdemultimedia/kioslave/audiocd/plugins/vorbis/Makefile.am	Thu Oct  7 14:16:44 2004
@@ -13,3 +13,6 @@
 pluginsdir = $(kde_datadir)/audiocd/plugins
 
 METASOURCES = AUTO
+
+messages: rc.cpp
+	$(XGETTEXT) *.cpp -o $(podir)/audiocd_encoder_vorbis.pot
diff -urN -x CVS kdemultimedia.orig/kioslave/audiocd/plugins/vorbis/audiocd_vorbis_encoder.kcfg kdemultimedia/kioslave/audiocd/plugins/vorbis/audiocd_vorbis_encoder.kcfg
--- kdemultimedia.orig/kioslave/audiocd/plugins/vorbis/audiocd_vorbis_encoder.kcfg	Sun Mar 21 04:59:00 2004
+++ kdemultimedia/kioslave/audiocd/plugins/vorbis/audiocd_vorbis_encoder.kcfg	Mon Oct  4 01:26:17 2004
@@ -29,7 +29,7 @@
     <entry name="vorbis_quality" type="Int">
       <label>Quality</label>
       <default>3</default>
-      <min>0</min>
+      <min>-1</min>
       <max>10</max>
     </entry>
    
diff -urN -x CVS kdemultimedia.orig/kmix/restore_kmix_volumes.desktop kdemultimedia/kmix/restore_kmix_volumes.desktop
--- kdemultimedia.orig/kmix/restore_kmix_volumes.desktop	Sat Sep 25 15:40:52 2004
+++ kdemultimedia/kmix/restore_kmix_volumes.desktop	Tue Oct  5 15:20:25 2004
@@ -27,9 +27,10 @@
 Name[pt]=Repor os Volumes
 Name[pt_BR]=Restaurar volumes do mixer
 Name[ru]=Восстановление настроек микшера
+Name[sk]=Obnoviť nastavenia mixéra
 Name[sl]=Obnovi nastavitve mešalnika
 Name[sr]=Обнови јачине миксете
-Name[sr@Latn]=Обнови јачине миксете
+Name[sr@Latn]=Obnovi jačine miksete
 Name[sv]=Återställ mixervolymer
 Name[ta]=ஒன்றுசேர்க்கும் ஒலியளவு மீட்கவும்
 Name[tg]=Аз Нав Захиракунии Баландиҳои Овозии Омехтакунак
diff -urN -x CVS kdemultimedia.orig/krec/kcm_krec.desktop kdemultimedia/krec/kcm_krec.desktop
--- kdemultimedia.orig/krec/kcm_krec.desktop	Sat Sep 25 15:40:54 2004
+++ kdemultimedia/krec/kcm_krec.desktop	Tue Oct  5 15:20:27 2004
@@ -61,9 +61,10 @@
 Comment[pt]=Configuração Global
 Comment[pt_BR]=Configuração global
 Comment[ru]=Общие параметры
+Comment[sk]=Globálne nastavenie
 Comment[sl]=Splošne nastavitve
 Comment[sr]=Опште подешавање
-Comment[sr@Latn]=Опште подешавање
+Comment[sr@Latn]=Opšte podešavanje
 Comment[sv]=Global inställning
 Comment[ta]=KRec இன் உலகலாவிய வடிவமைப்பு
 Comment[tg]=Танзимотҳои Саросарӣ
diff -urN -x CVS kdemultimedia.orig/krec/krec_exportitem.desktop kdemultimedia/krec/krec_exportitem.desktop
--- kdemultimedia.orig/krec/krec_exportitem.desktop	Tue Aug 31 09:22:07 2004
+++ kdemultimedia/krec/krec_exportitem.desktop	Mon Oct  4 15:25:33 2004
@@ -30,7 +30,7 @@
 Comment[sk]=Exportný modul pre KRec
 Comment[sl]=Izvozni vstavek za KRec
 Comment[sr]=KRec-ов прикључак за извоз
-Comment[sr@Latn]=KRec-ov dodatak za izvoz
+Comment[sr@Latn]=KRec-ov priključak za izvoz
 Comment[sv]=Exportinsticksmodul för Krec
 Comment[ta]=KRecற்கான ஏற்றுமதிசொருகுப்பொருள்
 Comment[tg]=Модули содирот барои KСабт
diff -urN -x CVS kdemultimedia.orig/libkcddb/kcmcddb/libkcddb.desktop kdemultimedia/libkcddb/kcmcddb/libkcddb.desktop
--- kdemultimedia.orig/libkcddb/kcmcddb/libkcddb.desktop	Sun Sep 12 15:56:35 2004
+++ kdemultimedia/libkcddb/kcmcddb/libkcddb.desktop	Tue Oct  5 15:20:29 2004
@@ -118,8 +118,9 @@
 Comment[pt]=Configurar a Transferência de CDDB
 Comment[pt_BR]=Configurar a recuperação do CDDB
 Comment[ru]=Настройка CDDB
+Comment[sk]=Nastavenie CDDB
 Comment[sr]=Подешавање добављања CDDB-а
-Comment[sr@Latn]=Подешавање добављања CDDB-а
+Comment[sr@Latn]=Podešavanje dobavljanja CDDB-a
 Comment[sv]=Anpassa hämtning från CDDB
 Comment[ta]=CDDB மீட்டெடுப்பை உள்ளமை
 Comment[tg]=Танзими Бозёбии CDDB
diff -urN -x CVS kdemultimedia.orig/noatun/modules/htmlexport/htmlexport.plugin kdemultimedia/noatun/modules/htmlexport/htmlexport.plugin
--- kdemultimedia.orig/noatun/modules/htmlexport/htmlexport.plugin	Sat Sep 25 15:40:57 2004
+++ kdemultimedia/noatun/modules/htmlexport/htmlexport.plugin	Tue Oct  5 15:20:32 2004
@@ -25,9 +25,10 @@
 Name[pt]=Exportação de Listas de Músicas em HTML
 Name[pt_BR]=Exportação de listas de reprodução em HTML
 Name[ru]=Экспорт списка песен в HTML
+Name[sk]=Export playlistu do HTML
 Name[sl]=Izvoz predvajalnega seznama v HTML
 Name[sr]=Извожење листе нумера у HTML
-Name[sr@Latn]=Извожење листе нумера у HTML
+Name[sr@Latn]=Izvoženje liste numera u HTML
 Name[sv]=HTML-spellistexport
 Name[ta]=HTML பாடல் பட்டியல் ஏற்றுமதி
 Name[tg]=HTML Содироти Рӯйхати Бозикуниҳо
diff -urN -x CVS kdemultimedia.orig/noatun/modules/infrared/infrared.plugin kdemultimedia/noatun/modules/infrared/infrared.plugin
--- kdemultimedia.orig/noatun/modules/infrared/infrared.plugin	Sat Sep 25 15:40:58 2004
+++ kdemultimedia/noatun/modules/infrared/infrared.plugin	Tue Oct  5 15:20:32 2004
@@ -79,9 +79,10 @@
 Comment[pt]=Controle o Noatun com um comando à distância
 Comment[pt_BR]=Controlar o Noatun com seu controle-remoto infravermelho
 Comment[ru]=Управление Noatun с помощью инфракрасного пульта
+Comment[sk]=Ovládajte Noatun vaším diaľkovým ovládačom
 Comment[sl]=Nadzorujte Noatun z vašim infrardečim daljinskim upravljalcem
 Comment[sr]=Контролишите Noatun вашим ИЦ даљинским управљачем
-Comment[sr@Latn]=Контролишите Noatun вашим ИЦ даљинским управљачем
+Comment[sr@Latn]=Kontrolišite Noatun vašim IC daljinskim upravljačem
 Comment[sv]=Styr Noatun med en infraröd fjärrkontroll
 Comment[ta]=உங்கள் IR கருவியால் Noatun இனைக் கட்டுப்படுத்தவும்
 Comment[tg]=Идоракунии Noatun бо IR дурдасти шумо
diff -urN -x CVS kdemultimedia.orig/noatun/modules/keyz/keyz.plugin kdemultimedia/noatun/modules/keyz/keyz.plugin
--- kdemultimedia.orig/noatun/modules/keyz/keyz.plugin	Sat Sep 25 15:40:58 2004
+++ kdemultimedia/noatun/modules/keyz/keyz.plugin	Tue Oct  5 15:20:32 2004
@@ -41,9 +41,10 @@
 Comment[pt]=Atalhos globais para a maioria das operações
 Comment[pt_BR]=Atalhos globais para a maioria das operações
 Comment[ru]=Привязки клавиш для большинства действий
+Comment[sk]=Globálne skratky pre väčšinu operácií
 Comment[sl]=Globalne bližnjice za večino operacij
 Comment[sr]=Глобалне пречице за већину операција
-Comment[sr@Latn]=Глобалне пречице за већину операција
+Comment[sr@Latn]=Globalne prečice za većinu operacija
 Comment[sv]=Globala snabbtangenter för de flesta operationerna
 Comment[ta]=பல செயலுகளுக்கான உலகளாவிய குறுக்குவழிகள்
 Comment[tg]=Миёнбурҳои саросарӣ барои бештари амалиётҳо
diff -urN -x CVS kdemultimedia.orig/noatun/modules/kjofol-skin/kjfont.cpp kdemultimedia/noatun/modules/kjofol-skin/kjfont.cpp
--- kdemultimedia.orig/noatun/modules/kjofol-skin/kjfont.cpp	Sun Feb  9 20:26:41 2003
+++ kdemultimedia/noatun/modules/kjofol-skin/kjfont.cpp	Mon Oct 18 15:40:53 2004
@@ -223,7 +223,8 @@
 //	kdDebug(66666) << k_funcinfo << "pixmap width=" << region.width() << endl;
 
 	// draw every char and add spacing in between these chars if defined
-	for ( unsigned int charPos=0; charPos < string.length(); charPos++ )
+	unsigned int stringLength(string.length());
+	for ( unsigned int charPos=0; charPos < stringLength; charPos++ )
 	{
 		char c = string[charPos]; // the character to be drawn next
 
diff -urN -x CVS kdemultimedia.orig/noatun/modules/kjofol-skin/kjofolui.plugin kdemultimedia/noatun/modules/kjofol-skin/kjofolui.plugin
--- kdemultimedia.orig/noatun/modules/kjofol-skin/kjofolui.plugin	Sat Sep 25 15:40:58 2004
+++ kdemultimedia/noatun/modules/kjofol-skin/kjofolui.plugin	Tue Oct  5 15:20:32 2004
@@ -35,9 +35,10 @@
 Comment[pt]=Leitor de aspectos do K-Jofol
 Comment[pt_BR]=Carregador de aparências (skins) para o K-Jofol
 Comment[ru]=Загрузчик образов K-Jofol
+Comment[sk]=Nahrávanie tém K-Jofol
 Comment[sl]=Nalagalnik za preobleke K-Jofol
 Comment[sr]=Учитавач кошуљица за К-Jofol кошуљице
-Comment[sr@Latn]=Учитавач кошуљица за К-Jofol кошуљице
+Comment[sr@Latn]=Učitavač košuljica za K-Jofol košuljice
 Comment[sv]=Skalladdare för K-Jofol-skal
 Comment[ta]=கே-ஜோபோல் அலங்கார அமைப்புக்கான ஏற்றி
 Comment[tg]=Боркунандаи намуд барои намудҳои K-Jofol
diff -urN -x CVS kdemultimedia.orig/noatun/modules/marquis/marquis.plugin kdemultimedia/noatun/modules/marquis/marquis.plugin
--- kdemultimedia.orig/noatun/modules/marquis/marquis.plugin	Tue Aug 24 08:30:44 2004
+++ kdemultimedia/noatun/modules/marquis/marquis.plugin	Mon Oct  4 15:25:39 2004
@@ -60,7 +60,7 @@
 Comment[sk]=Modul pre spoluprácu so Správcom sedenia
 Comment[sl]=Vstavek za delovanje z Upravljalnikom sej
 Comment[sr]=Прикључак за интеракцију са менаџером сесије
-Comment[sr@Latn]=Dodatak za interakciju sa menadžerom sesije
+Comment[sr@Latn]=Priključak za interakciju sa menadžerom sesije
 Comment[sv]=Insticksprogram för att samverka med sessionshanteraren
 Comment[ta]=அமர்வு மேலாளருடன் தொடர்பாடும் சொருகுப்பொருள்
 Comment[tg]=Модул барои амали мутақобила бо Мудири Сеанс
diff -urN -x CVS kdemultimedia.orig/noatun/modules/splitplaylist/playlist.cpp kdemultimedia/noatun/modules/splitplaylist/playlist.cpp
--- kdemultimedia.orig/noatun/modules/splitplaylist/playlist.cpp	Sun Dec 28 17:33:15 2003
+++ kdemultimedia/noatun/modules/splitplaylist/playlist.cpp	Mon Oct  4 10:58:40 2004
@@ -80,7 +80,9 @@
 	}
 
 	if (!nextItem) // don't set a null-item as current item
-		return 0;
+	{
+		nextItem = static_cast<SafeListViewItem*>(static_cast<SafeListViewItem*>(getFirst().data()));
+	}
 
 	PlaylistItem oldCurrent = currentItem;
 	setCurrent(nextItem, play);
diff -urN -x CVS kdemultimedia.orig/noatun/modules/voiceprint/voiceprint.plugin kdemultimedia/noatun/modules/voiceprint/voiceprint.plugin
--- kdemultimedia.orig/noatun/modules/voiceprint/voiceprint.plugin	Sat Sep 25 15:40:58 2004
+++ kdemultimedia/noatun/modules/voiceprint/voiceprint.plugin	Tue Oct  5 15:20:33 2004
@@ -62,9 +62,10 @@
 Comment[pt]=Um visualizador do comportamento vocal
 Comment[pt_BR]=Um visualizador voiceprint
 Comment[ru]=визуализация voiceprint
+Comment[sk]=Vizualizátor voiceprint
 Comment[sl]=Predstavitelj glasovnega zapisa
 Comment[sr]=Визуализатор гласовног отиска
-Comment[sr@Latn]=Визуализатор гласовног отиска
+Comment[sr@Latn]=Vizualizator glasovnog otiska
 Comment[sv]=En röstavtrycksvisare
 Comment[ta]=குரல் அச்சு படக்காட்சி
 Comment[tg]=Дидорсози voiceprint
diff -urN -x CVS kdemultimedia.orig/noatun/modules/winskin/vis/visQueue.cpp kdemultimedia/noatun/modules/winskin/vis/visQueue.cpp
--- kdemultimedia.orig/noatun/modules/winskin/vis/visQueue.cpp	Sun Aug 29 23:12:42 2004
+++ kdemultimedia/noatun/modules/winskin/vis/visQueue.cpp	Mon Oct 18 22:38:29 2004
@@ -16,23 +16,24 @@
 
 VISQueue::VISQueue(int elements) {
   this->elements=elements;
-  int i;
-  visArrayQueue.reserve(elements);
-  for(i=0;i<elements;i++) {
+  
+  visArrayQueue = new std::vector<float>*[elements];
+  for(int i=0;i<elements;i++)
+  {
     visArrayQueue[i]=new std::vector<float>;
   }
 
 }
 
 VISQueue::~VISQueue() {
-  int i;
-
-  for(i=0;i<elements;i++) {
+  for(int i=0;i<elements;i++) {
     delete visArrayQueue[i];
   }
+  delete [] visArrayQueue;
 }
 
-std::vector<float>* VISQueue::getElement(int i) {
+std::vector<float>* VISQueue::getElement(int i)
+{
   if ( (i < 0) || (i>elements) ) {
     return visArrayQueue[0];
   }
diff -urN -x CVS kdemultimedia.orig/noatun/modules/winskin/vis/visQueue.h kdemultimedia/noatun/modules/winskin/vis/visQueue.h
--- kdemultimedia.orig/noatun/modules/winskin/vis/visQueue.h	Wed Jul 30 06:21:54 2003
+++ kdemultimedia/noatun/modules/winskin/vis/visQueue.h	Mon Oct 18 22:38:29 2004
@@ -20,7 +20,7 @@
 class VISQueue {
 
   int elements;
-  std::vector< std::vector<float>* > visArrayQueue; 
+  std::vector<float> **visArrayQueue; 
 
  public:
   VISQueue(int elements);
diff -urN -x CVS kdemultimedia.orig/noatun/modules/winskin/waSkin.cpp kdemultimedia/noatun/modules/winskin/waSkin.cpp
--- kdemultimedia.orig/noatun/modules/winskin/waSkin.cpp	Sat Jun 12 23:00:11 2004
+++ kdemultimedia/noatun/modules/winskin/waSkin.cpp	Mon Oct  4 12:25:52 2004
@@ -59,8 +59,6 @@
 
 WaSkin::WaSkin() : QWidget(0, "NoatunWinampSkin"), UserInterface()
 {
-    NOATUNPLUGINC(WaSkin);
-
     mJumpPressed = false;
     mVolumePressed = false;
     mBalancePressed = false;
diff -urN -x CVS kdemultimedia.orig/noatun/modules/winskin/winSkinConfig.cpp kdemultimedia/noatun/modules/winskin/winSkinConfig.cpp
--- kdemultimedia.orig/noatun/modules/winskin/winSkinConfig.cpp	Mon Feb 24 16:04:53 2003
+++ kdemultimedia/noatun/modules/winskin/winSkinConfig.cpp	Mon Oct  4 12:25:52 2004
@@ -18,6 +18,7 @@
 #include <kurlrequesterdlg.h>
 #include <qdir.h>
 
+#include "waSkin.h"
 #include "waSkinManager.h"
 #include "winSkinConfig.h"
 
@@ -64,7 +65,7 @@
     config->sync();
 
     if (skin_list->currentText() != orig_skin) {
-        mWaSkinManager->loadSkin(skin_list->currentText());
+        _waskin_instance->loadSkin(skin_list->currentText());
         orig_skin = skin_list->currentText();
     }
 }
diff -urN -x CVS kdemultimedia.orig/noatun/modules/winskin/winskin.plugin kdemultimedia/noatun/modules/winskin/winskin.plugin
--- kdemultimedia.orig/noatun/modules/winskin/winskin.plugin	Sat Sep 25 15:40:59 2004
+++ kdemultimedia/noatun/modules/winskin/winskin.plugin	Tue Oct  5 15:20:33 2004
@@ -82,9 +82,10 @@
 Comment[pt]=Um leitor de aspectos do WinAmp
 Comment[pt_BR]=Um carregador de aparências (skins) do Winamp
 Comment[ru]=Загрузчик тем Winamp
+Comment[sk]=Načítavač tém WinAmp
 Comment[sl]=Nalagalnik preoblek Winamp
 Comment[sr]=Учитавач Winamp-ових кошуљица
-Comment[sr@Latn]=Учитавач Winamp-ових кошуљица
+Comment[sr@Latn]=Učitavač Winamp-ovih košuljica
 Comment[sv]=Skalladdare för Winamp
 Comment[ta]=வின் ஆம்ப் அலங்கார அமைப்பு ஏற்றி
 Comment[tg]=Боркунандаи намуди Winamp
