--- kdemultimedia-3.0.3/noatun/library/engine.cpp--	2002-09-05 18:44:19.000000000 +0200
+++ kdemultimedia-3.0.3/noatun/library/engine.cpp	2002-09-05 18:45:19.000000000 +0200
@@ -180,9 +180,11 @@ namespace VolumeControls
 		else
 		{
 #endif
+#if 0
 			if (Arts::CpuInfo::flags()&Arts::CpuInfo::CpuSSE)
 				return new SoftwareSSE(e);
 			else
+#endif
 				return new Software(e);
 
 #ifdef HARDWARE_VOLUME
