--- kdemultimedia-3.0.3/arts/builder/createtool.cpp--	2002-08-28 16:52:43.000000000 +0200
+++ kdemultimedia-3.0.3/arts/builder/createtool.cpp	2002-08-28 16:53:17.000000000 +0200
@@ -99,6 +99,7 @@ void CreateTool::mouseMoveEvent(QMouseEv
 	p->drawRect(componentRect);
 	p->restore();
 	p->end();
+	delete p;
 }
 
 void CreateTool::mouseReleaseEvent(QMouseEvent *e)
--- kdemultimedia-3.0.3/arts/builder/module.cpp--	2002-08-28 16:52:00.000000000 +0200
+++ kdemultimedia-3.0.3/arts/builder/module.cpp	2002-08-28 16:52:36.000000000 +0200
@@ -189,6 +189,7 @@ Module::~Module()
 
 	StructureDesc.freeModuleDesc(ModuleDesc);
 	arts_debug("ok...");
+	delete _pixmap;
 }
 
 bool Module::moveInternal(int x, int y)
--- kdemultimedia-3.0.3/arts/builder/structureport.cpp--	2002-08-28 16:51:32.000000000 +0200
+++ kdemultimedia-3.0.3/arts/builder/structureport.cpp	2002-08-28 16:51:51.000000000 +0200
@@ -61,6 +61,7 @@ StructurePort::~StructurePort()
 {
 	hide();
 	StructureDesc.freeStructurePortDesc(SPortDesc);
+	delete _port;
 }
 
 bool StructurePort::moveInternal(int x, int y)
