--- kdemultimedia-3.3.0/mpeglib/lib/input/cdromAccess.cpp	2004-08-23 18:32:32.001517760 +0200
+++ kdemultimedia-3.3.0/mpeglib/lib/input/cdromAccess.cpp.orig	2004-08-23 18:32:26.974282016 +0200
@@ -27,7 +27,7 @@
   #include <sys/types.h>
   #ifndef __u64
   #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,70)
-
+  typedef unsigned long long __u64;
   #endif
   #endif
   #include "cdromAccess_Linux.cpp"
