Index: CDDBSetup.cpp
===================================================================
RCS file: /home/kde/kdemultimedia/kscd/CDDBSetup.cpp,v
retrieving revision 1.33
diff -u -p -r1.33 CDDBSetup.cpp
--- kdemultimedia/kscd/CDDBSetup.cpp	2001/07/27 12:12:09	1.33
+++ kdemultimedia/kscd/CDDBSetup.cpp	2002/02/03 09:50:06
@@ -8,7 +8,7 @@
 #include <stdio.h>
 
 #include <qvalidator.h>
-
+#include <knumvalidator.h>
 #include <kglobal.h>
 #include <kstddirs.h>
 #include <kapp.h>
@@ -63,8 +63,7 @@ CDDBSetup::CDDBSetup
     connect(currentSubmitAddPB, SIGNAL(clicked()), this, SLOT(insertSUL()));
     connect(currentSubmitDelPB, SIGNAL(clicked()), this, SLOT(removeSUL()));
 
-    QIntValidator *port_validator=new QIntValidator(this,"HTTP Port Validator");
-    port_validator->setRange(0, INT_MAX );
+    KIntValidator *port_validator=new KIntValidator(0,INT_MAX,proxy_port_ef);
     proxy_port_ef->setValidator(port_validator);
 
     proxy_port_ef->setEnabled(cddb_http_cb->isChecked());
@@ -241,7 +240,7 @@ CDDBSetup::set_defaults()
     remote_cddb_cb->setChecked(true);
     cddb_http_cb->setChecked(false);
     // Leave proxy host and port values unchanged, just disable them
-
+    cddb_timeout_ef->setText("30");
     emit updateCurrentServer();
 } // set_defaults
 
Index: CDDBSetupData.cpp
===================================================================
RCS file: /home/kde/kdemultimedia/kscd/CDDBSetupData.cpp,v
retrieving revision 1.23
diff -u -p -r1.23 CDDBSetupData.cpp
--- kdemultimedia/kscd/CDDBSetupData.cpp	2001/07/27 09:09:01	1.23
+++ kdemultimedia/kscd/CDDBSetupData.cpp	2002/02/03 09:50:07
@@ -55,7 +55,7 @@ CDDBSetupData::CDDBSetupData
 	cddb_timeout_ef->setText( "30" );
 	cddb_timeout_ef->setMaxLength ( 5 );
 	cddb_timeout_ef->setFixedWidth ( 5*fm.maxWidth() );
-        cddb_timeout_ef->setValidator(new QIntValidator (cddb_timeout_ef));
+        cddb_timeout_ef->setValidator(new KIntValidator (0,9999,cddb_timeout_ef));
 	lay3->addWidget ( cddb_timeout_ef );
 	QLabel* cddb_timeout_lb = new QLabel( i18n("seconds freedb timeout"), group, "CDDBTimeoutLabel" );
         lay3->addWidget ( cddb_timeout_lb );
