Index: kaboodle/kaboodleengine.desktop
===================================================================
--- kaboodle/kaboodleengine.desktop	(.../tags/KDE/3.4.2/kdemultimedia)	(revision 442272)
+++ kaboodle/kaboodleengine.desktop	(.../branches/KDE/3.4/kdemultimedia)	(revision 442272)
@@ -3,6 +3,7 @@
 Name=Embedded Media Player Engine
 Name[af]=Ingebedde Media Speler Masjien
 Name[bn]=সন্নিবিষ্ট মিডিয়া প্লেয়ার ইঞ্জিন
+Name[br]=Keflusker enframmet soner liesvedia
 Name[bs]=Uključeni Media Player
 Name[ca]=Motor del reproductor multimèdia incrustat
 Name[cs]=Vestavěný přehrávač médií
Index: kaudiocreator/encodefileimp.cpp
===================================================================
--- kaudiocreator/encodefileimp.cpp	(.../tags/KDE/3.4.2/kdemultimedia)	(revision 442272)
+++ kaudiocreator/encodefileimp.cpp	(.../branches/KDE/3.4/kdemultimedia)	(revision 442272)
@@ -32,6 +32,8 @@
   genre->insertStringList(m_genres.i18nList());
   // Specify to only accept wav files
   file->setFilter("*.wav|Wav Files");
+
+  connect(encodeButton,SIGNAL(clicked()),this,SLOT(encode()));
 }
 
 /**
@@ -53,6 +55,9 @@
   newJob->track = track->value();
 
   newJob->track_title = track_title->text();
+  if ((track_artist->text()).isEmpty())
+     newJob->track_artist = artist->text();
+  else
   newJob->track_artist = track_artist->text();
   newJob->track_comment = track_comment->text();
 
Index: kaudiocreator/encodefile.ui
===================================================================
--- kaudiocreator/encodefile.ui	(.../tags/KDE/3.4.2/kdemultimedia)	(revision 442272)
+++ kaudiocreator/encodefile.ui	(.../branches/KDE/3.4/kdemultimedia)	(revision 442272)
@@ -1,4 +1,4 @@
-<!DOCTYPE UI><UI version="3.2" stdsetdef="1">
+<!DOCTYPE UI><UI version="3.3" stdsetdef="1">
 <class>EncodeFile</class>
 <widget class="KDialog">
     <property name="name">
@@ -9,7 +9,7 @@
             <x>0</x>
             <y>0</y>
             <width>339</width>
-            <height>435</height>
+            <height>452</height>
         </rect>
     </property>
     <property name="caption">
@@ -292,7 +292,7 @@
                 </spacer>
                 <widget class="QPushButton">
                     <property name="name">
-                        <cstring>pushButton6</cstring>
+                        <cstring>encodeButton</cstring>
                     </property>
                     <property name="text">
                         <string>&amp;Encode File</string>
@@ -300,7 +300,7 @@
                 </widget>
                 <widget class="QPushButton">
                     <property name="name">
-                        <cstring>pushButton5</cstring>
+                        <cstring>closeButton</cstring>
                     </property>
                     <property name="text">
                         <string>&amp;Close</string>
@@ -310,9 +310,11 @@
         </widget>
     </vbox>
 </widget>
+<customwidgets>
+</customwidgets>
 <connections>
     <connection>
-        <sender>pushButton5</sender>
+        <sender>closeButton</sender>
         <signal>clicked()</signal>
         <receiver>EncodeFile</receiver>
         <slot>accept()</slot>
@@ -329,11 +331,12 @@
     <tabstop>year</tabstop>
     <tabstop>genre</tabstop>
     <tabstop>comment</tabstop>
-    <tabstop>pushButton6</tabstop>
-    <tabstop>pushButton5</tabstop>
+    <tabstop>encodeButton</tabstop>
+    <tabstop>closeButton</tabstop>
 </tabstops>
 <layoutdefaults spacing="6" margin="11"/>
 <includehints>
+    <includehint>kdialog.h</includehint>
     <includehint>kurlrequester.h</includehint>
     <includehint>klineedit.h</includehint>
     <includehint>kpushbutton.h</includehint>
Index: Makefile.cvs
===================================================================
--- Makefile.cvs	(.../tags/KDE/3.4.2/kdemultimedia)	(revision 0)
+++ Makefile.cvs	(.../branches/KDE/3.4/kdemultimedia)	(revision 442272)
@@ -0,0 +1,16 @@
+
+all: 
+	@echo "This Makefile is only for the CVS repository"
+	@echo "This will be deleted before making the distribution"
+	@echo ""
+	@if test ! -d admin; then \
+	   echo "Please recheckout this module!" ;\
+	   echo "for cvs: use checkout once and after that update again" ;\
+	   echo "for cvsup: checkout kde-common from cvsup and" ;\
+	   echo "   link kde-common/admin to ./admin" ;\
+	   exit 1 ;\
+	fi
+	$(MAKE) -f admin/Makefile.common cvs
+
+.SILENT:
+

Property changes on: Makefile.cvs
___________________________________________________________________
Name: svn:keywords
   + Author Date Id Revision

Index: kmix/kmixdockwidget.cpp
===================================================================
--- kmix/kmixdockwidget.cpp	(.../tags/KDE/3.4.2/kdemultimedia)	(revision 442272)
+++ kmix/kmixdockwidget.cpp	(.../branches/KDE/3.4/kdemultimedia)	(revision 442272)
@@ -86,10 +86,12 @@
   if ( a ) a->plug( popupMenu );
 
   // Put "Select Master Channel" dialog in context menu
+  if ( m_mixer != 0 ) {
   (void)new KAction( i18n("Select Channel"), 0, this, SLOT(selectMaster()),
   actionCollection(), "select_master");
   KAction *a2 = actionCollection()->action( "select_master" );
   if (a2) a2->plug( popupMenu );
+  }
 
    // Setup volume preview
   if ( _playBeepOnVolumeChange ) {
Index: kmix/version.h
===================================================================
--- kmix/version.h	(.../tags/KDE/3.4.2/kdemultimedia)	(revision 442272)
+++ kmix/version.h	(.../branches/KDE/3.4/kdemultimedia)	(revision 442272)
@@ -18,4 +18,4 @@
  * License along with this program; if not, write to the Free
  * Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
  */
-#define APP_VERSION "2.6"
+#define APP_VERSION "2.6.1"
Index: kioslave/audiocd/plugins/lame/encoderlame.cpp
===================================================================
--- kioslave/audiocd/plugins/lame/encoderlame.cpp	(.../tags/KDE/3.4.2/kdemultimedia)	(revision 442272)
+++ kioslave/audiocd/plugins/lame/encoderlame.cpp	(.../branches/KDE/3.4/kdemultimedia)	(revision 442272)
@@ -244,6 +244,7 @@
 #else
    QStringList libpaths, libnames;
    libpaths << QFL1("/usr/lib/")
+	    << QFL1("/usr/lib64/")
             << QFL1("/usr/local/lib/")
             << QString::null;
 
Index: kioslave/audiocd/audiocd.cpp
===================================================================
--- kioslave/audiocd/audiocd.cpp	(.../tags/KDE/3.4.2/kdemultimedia)	(revision 442272)
+++ kioslave/audiocd/audiocd.cpp	(.../branches/KDE/3.4/kdemultimedia)	(revision 442272)
@@ -366,6 +366,7 @@
 		kdDebug(7117) << "cdda_open failed" << endl;
 
 		error(KIO::ERR_CANNOT_OPEN_FOR_READING, url.path());
+		cdda_close(drive);
 		return 0;
 	}
 

Property changes on: .
___________________________________________________________________
Name: svn:externals
   + admin https://svn.kde.org/home/kde/branches/KDE/3.4/kde-common/admin


