diff -urN kdemultimedia-3.0.9.orig/kscd/kscd.cpp kdemultimedia-3.0.9/kscd/kscd.cpp
--- kdemultimedia-3.0.9.orig/kscd/kscd.cpp	Mon Oct 28 10:59:49 2002
+++ kdemultimedia-3.0.9/kscd/kscd.cpp	Thu Oct 31 23:38:48 2002
@@ -279,7 +279,7 @@
   action->plugAccel(accels);
   action = new KAction(i18n("Decrease Volume"), Key_Minus, this, SLOT(decVolume()), this, "DecVolume");
   action->plugAccel(accels);
-  action = new KAction(i18n("Options"), CTRL + Key_T, this, SLOT(showConfig), this);
+  action = new KAction(i18n("Options"), CTRL + Key_T, this, SLOT(showConfig), this, "Options");
   action->plugAccel(accels);
   action = new KAction(i18n("Shuffle"), Key_R, this, SLOT(randomSelected()), this, "Shuffle");
   action->plugAccel(accels);
