Index: krec/krec_exportitem.desktop
===================================================================
--- krec/krec_exportitem.desktop	(.../tags/KDE/3.5.4/kdemultimedia)	(revision 570934)
+++ krec/krec_exportitem.desktop	(.../branches/KDE/3.5/kdemultimedia)	(revision 570934)
@@ -26,7 +26,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​ការ​នាំចេញ​សម្រាប់ KRec
 Comment[lt]=Eksporto priedas, skirtas KRec
 Comment[mk]=Приклучок за изнесување за KRec
-Comment[nb]=Tilleggsprogram for eksport i KRec
+Comment[nb]=Programtillegg for eksport i KRec
 Comment[nl]=Exportplugin voor KRec
 Comment[nn]=Eksporttillegg for KRec
 Comment[pl]=Wtyczka eksportu dla KRec
Index: kscd/kscd.desktop
===================================================================
--- kscd/kscd.desktop	(.../tags/KDE/3.5.4/kdemultimedia)	(revision 570934)
+++ kscd/kscd.desktop	(.../branches/KDE/3.5/kdemultimedia)	(revision 570934)
@@ -24,7 +24,7 @@
 GenericName[hu]=CD-lejátszó
 GenericName[is]=CD spilari
 GenericName[it]=Lettore CD
-GenericName[ja]=CDプレーヤ
+GenericName[ja]=CD プレーヤ
 GenericName[km]=កម្មវិធី​ចាក់​ស៊ីឌី
 GenericName[lt]=CD grotuvas
 GenericName[lv]=CD Atskaņotājs
Index: kaudiocreator/audiocd_extract.desktop
===================================================================
--- kaudiocreator/audiocd_extract.desktop	(.../tags/KDE/3.5.4/kdemultimedia)	(revision 570934)
+++ kaudiocreator/audiocd_extract.desktop	(.../branches/KDE/3.5/kdemultimedia)	(revision 570934)
@@ -21,7 +21,7 @@
 Name[hu]=Hangsávok kinyerése és tömörítése
 Name[is]=Afrita og kóða hljóðspor
 Name[it]=Estrai e comprimi le tracce audio
-Name[ja]=オーディオトラックを取りだしエンコード
+Name[ja]=オーディオトラックを取り出しエンコード
 Name[km]=ស្រង់ចេញ និង អ៊ិនកូដ​បទ​អូឌីយ៉ូ
 Name[lt]=Nuskaityti ir įkoduoti audio takelius
 Name[nb]=Pakk ut og dekod lydfiler
Index: kaudiocreator/eventsrc
===================================================================
--- kaudiocreator/eventsrc	(.../tags/KDE/3.5.4/kdemultimedia)	(revision 570934)
+++ kaudiocreator/eventsrc	(.../branches/KDE/3.5/kdemultimedia)	(revision 570934)
@@ -66,7 +66,7 @@
 Comment[hu]=A sorban várakozó feladatok befejeződtek
 Comment[is]=Öllum verkum í biðröð hefur verið lokið
 Comment[it]=Tutti i processi in coda sono terminati
-Comment[ja]=キューにある全ジョブが終了
+Comment[ja]=キューにある全ジョブが完了しました
 Comment[km]=ការងារ​ទាំងអស់​នៅ​ក្នុង​ជួរ ត្រូវ​បាន​បញ្ចប់​ហើយ
 Comment[lt]=Visi eilėje buvę darbai užbaigti
 Comment[mk]=Сите задачи во редицата завршија
@@ -109,7 +109,7 @@
 Name[hu]=Hangsáv kimásolva
 Name[is]=Lag afritað
 Name[it]=Track compressa
-Name[ja]=トラックリッピング
+Name[ja]=トラックのリッピング終了
 Name[km]=បាន​ច្រៀក​បទ
 Name[lt]=Takelio nuskaitymas baigtas
 Name[mk]=Спуштена песна
@@ -196,7 +196,7 @@
 Name[hu]=CD kimásolva
 Name[is]=Diskur afritaður
 Name[it]=CD estratto
-Name[ja]=CD リッパー
+Name[ja]=CDのリッピング終了
 Name[km]=បាន​ច្រៀក​ស៊ីឌី
 Name[lt]=CD nuskaitytas
 Name[mk]=CD-то е спуштено
@@ -285,7 +285,7 @@
 Name[hu]=Hangsáv kódolása kész
 Name[is]=Lag kóðað
 Name[it]=Traccia codificata
-Name[ja]=トラックエンコード
+Name[ja]=トラックのエンコード終了
 Name[km]=បាន​អ៊ិនកូដ​បទ
 Name[lt]=Takelis įkoduotas
 Name[mk]=Кодирање на песната
@@ -373,7 +373,7 @@
 Name[hu]=CD kódolása kész
 Name[is]=Diskur kóðaður
 Name[it]=CD codificato
-Name[ja]=CDエンコード
+Name[ja]=CDのエンコード終了
 Name[km]=បាន​អ៊ិនកូដ​ស៊ីឌី
 Name[lt]=CD įkoduotas
 Name[mk]=Кодирано CD
Index: kmix/mdwslider.cpp
===================================================================
--- kmix/mdwslider.cpp	(.../tags/KDE/3.5.4/kdemultimedia)	(revision 570934)
+++ kmix/mdwslider.cpp	(.../branches/KDE/3.5/kdemultimedia)	(revision 570934)
@@ -281,14 +281,12 @@
 		 }
 
 		 // create labels to hold volume values (taken from qamix/kamix)
-		 QLabel *number = new QLabel( this );
+		 QLabel *number = new QLabel( "100", this );
 		 slinumLayout->addWidget( number );
-		 QFontMetrics *qfm = new QFontMetrics( number->font() );
 		 number->setFrameStyle( QFrame::Panel | QFrame::Sunken );
-		 number->setMinimumWidth( qfm->width("MMM", 3) );
-		 number->setLineWidth( 3 );
+		 number->setLineWidth( 2 );
+		 number->setMinimumWidth( number->sizeHint().width() );
 		 number->setPaletteBackgroundColor( QColor(190, 250, 190) );
-		 delete qfm;
 		 // don't show the value by default
 		 number->hide();
 		 updateValue( number, chid );

Property changes on: .
___________________________________________________________________
Name: svn:externals
   + admin https://svn.kde.org/home/kde/branches/KDE/3.5/kde-common/admin


