--- kdemultimedia-3.3.91/mpeglib/lib/input/cdromAccess.cpp.orig	2005-01-03 15:53:36.000000000 +0100
+++ kdemultimedia-3.3.91/mpeglib/lib/input/cdromAccess.cpp	2005-01-24 17:21:57.197144304 +0100
@@ -25,14 +25,12 @@
 #ifdef OS_Linux
   #include <linux/version.h>
   #include <sys/types.h>
-  #ifndef __u64
-  #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,70)
+  #if !defined(LIBC_HEADERS_VERSION) && (LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,70))
   #include <bits/wordsize.h>
   #if __WORDSIZE == 64
   typedef unsigned long __u64;
   #else
   typedef unsigned long long __u64;
-  #endif 
   #endif
   #endif
   #include "cdromAccess_Linux.cpp"
