--- kdemultimedia-3.1.3/kioslave/audiocd/audiocd.cpp.cdrom	Wed May  7 13:46:47 2003
+++ kdemultimedia-3.1.3/kioslave/audiocd/audiocd.cpp	Fri Sep  5 12:26:01 2003
@@ -27,14 +27,6 @@
 #include <unistd.h>
 #include <stdlib.h>
 
-#include <qfile.h>
-#include <qstrlist.h>
-#include <qdatetime.h>
-#include <qregexp.h>
-
-typedef Q_INT16 size16;
-typedef Q_INT32 size32;
-
 extern "C"
 {
 #include <cdda_interface.h>
@@ -72,6 +64,14 @@
 
 void paranoiaCallback(long, int);
 }
+#include <qfile.h>
+#include <qstrlist.h>
+#include <qdatetime.h>
+#include <qregexp.h>
+
+typedef Q_INT16 size16;
+typedef Q_INT32 size32;
+
 #include <kconfig.h>
 #include <kdebug.h>
 #include <kurl.h>
--- kdemultimedia-3.1.3/kioslave/audiocd/Makefile.am.cdrom	Wed May  7 13:46:47 2003
+++ kdemultimedia-3.1.3/kioslave/audiocd/Makefile.am	Fri Sep  5 13:58:31 2003
@@ -6,7 +6,7 @@
 
 kde_module_LTLIBRARIES = kio_audiocd.la
 
-kio_audiocd_la_SOURCES = cddb.cpp audiocd.cpp
+kio_audiocd_la_SOURCES = audiocd.cpp cddb.cpp
 
 kio_audiocd_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) $(CDPARANOIA_RPATH) \
  $(LAME_RPATH) $(VORBIS_RPATH) -module $(KDE_PLUGIN)
--- kdemultimedia-3.1.3/kaudiocreator/cdconfigimp.cpp.cdrom	Wed May  7 13:46:47 2003
+++ kdemultimedia-3.1.3/kaudiocreator/cdconfigimp.cpp	Fri Sep  5 12:22:38 2003
@@ -18,12 +18,25 @@
   Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */
 
+#include <config.h>
+
+// Many Linux systems do not define LINUX. It's something else that I
+// can't be bothered to look up just right now.
+#if defined(LINUX) || defined(__linux__)
+extern "C"
+{
+/* This is in support for the Mega Hack, if cdparanoia ever is fixed, or we
+   use another ripping library we can remove this.  */
+#include <sys/ioctl.h>
+#include <linux/cdrom.h>
+
+}
+#endif
+
 #include <qpushbutton.h>
 #include <kprocess.h>
 #include <kmessagebox.h>
 
-#include <config.h>
-
 // Moved the HAVE_LAME define in order to avoid redefined-warnings.
 #ifndef HAVE_LAME
 #define HAVE_LAME 0
@@ -68,19 +81,6 @@
 #include <cdda_paranoia.h>
 }
 
-// Many Linux systems do not define LINUX. It's something else that I
-// can't be bothered to look up just right now.
-#if defined(LINUX) || defined(__linux__)
-extern "C"
-{
-/* This is in support for the Mega Hack, if cdparanoia ever is fixed, or we
-   use another ripping library we can remove this.  */
-#include <linux/cdrom.h>
-#include <sys/ioctl.h>
-
-}
-#endif
-
 #include <kdebug.h>
 #include <kurl.h>
 #include <kprotocolmanager.h>
--- kdemultimedia-3.1.3/kaudiocreator/Makefile.am.cdrom	Wed May  7 13:46:47 2003
+++ kdemultimedia-3.1.3/kaudiocreator/Makefile.am	Fri Sep  5 13:10:34 2003
@@ -2,7 +2,7 @@
 
 bin_PROGRAMS = kaudiocreator
 
-kaudiocreator_SOURCES = kmacroexpander.cpp cddb.cpp main.cpp kaudiocreator.cpp cdconfigimp.cpp  encoderconfigimp.cpp  job.cpp  queconfigimp.cpp  ripconfigimp.cpp  tracksconfigimp.cpp  cdconfig.ui  encoderconfig.ui  queconfig.ui  ripconfig.ui  tracksconfig.ui id3tagdialog.ui wizard.ui
+kaudiocreator_SOURCES = cdconfigimp.cpp kmacroexpander.cpp cddb.cpp main.cpp kaudiocreator.cpp encoderconfigimp.cpp  job.cpp  queconfigimp.cpp  ripconfigimp.cpp  tracksconfigimp.cpp  cdconfig.ui  encoderconfig.ui  queconfig.ui  ripconfig.ui  tracksconfig.ui id3tagdialog.ui wizard.ui
 
 EXTRA_DIST = kaudiocreator.h kaudiocreator.desktop cdconfigimp.h  encoderconfigimp.h  queconfigimp.h  ripconfigimp.h  tracksconfigimp.h cddb.h
 
--- kdemultimedia-3.1.3/admin/acinclude.m4.in.cdrom	Fri Sep  5 12:35:45 2003
+++ kdemultimedia-3.1.3/admin/acinclude.m4.in	Fri Sep  5 13:12:22 2003
@@ -2840,7 +2840,9 @@
             CXXFLAGS="-ansi -D_XOPEN_SOURCE=500 -D_BSD_SOURCE -Wcast-align -Wconversion $CXXFLAGS"
           ;;
         esac
-        CXXFLAGS="-Wall -pedantic -W -Wpointer-arith -Wwrite-strings $CXXFLAGS"
+        CXXFLAGS="-Wall -W -Wpointer-arith -Wwrite-strings $CXXFLAGS"
+	 dnl -pedantic + g++ 3.3 + <linux/cdrom.h> = causes
+	 dnl "error: ISO C++ forbids braced-groups within expressions" in <linux/byteorder/swab.h>
         KDE_CHECK_COMPILER_FLAG(Wundef,[CXXFLAGS="-Wundef $CXXFLAGS"])
         KDE_CHECK_COMPILER_FLAG(Wno-long-long,[CXXFLAGS="-Wno-long-long $CXXFLAGS"])
         KDE_CHECK_COMPILER_FLAG(Wnon-virtual-dtor,[CXXFLAGS="-Wnon-virtual-dtor $CXXFLAGS"])
