diff -urN -x CVS kdemultimedia.orig/arts/modules/common/Makefile.am kdemultimedia/arts/modules/common/Makefile.am
--- kdemultimedia.orig/arts/modules/common/Makefile.am	Tue Aug 26 10:43:25 2003
+++ kdemultimedia/arts/modules/common/Makefile.am	Wed Aug 18 07:54:54 2004
@@ -19,6 +19,7 @@
 	env_context_impl.cc env_effectrackitem_impl.cc \
 	env_instrumentitem_impl.cc env_item_impl.cc \
 	env_mixeritem_impl.cc
+libartsmodulescommon_la_COMPILE_FIRST = artsmodulescommon.h
 
 libartsmodulescommon_la_LIBADD = \
 	$(top_builddir)/arts/gui/common/libartsgui_idl.la \
diff -urN -x CVS kdemultimedia.orig/arts/modules/synth/Makefile.am kdemultimedia/arts/modules/synth/Makefile.am
--- kdemultimedia.orig/arts/modules/synth/Makefile.am	Tue Aug 26 10:43:25 2003
+++ kdemultimedia/arts/modules/synth/Makefile.am	Wed Aug 18 07:55:23 2004
@@ -25,6 +25,7 @@
 	synth_capture_wav_impl.cc \
 	synth_midi_test_impl.cc synth_sequence_impl.cc synth_sequence_freq_impl.cc \
 	synth_midi_debug_impl.cc objectcache_impl.cc synth_nil_impl.cc synth_debug_impl.cc synth_data_impl.cc
+libartsmodulessynth_la_COMPILE_FIRST = artsmodulessynth.h
 
 libartsmodulessynth_la_LIBADD = \
 	$(top_builddir)/arts/runtime/libartsbuilder.la \
